{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2687cc20",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: rhdf5\n",
      "\n",
      "Loading required package: magrittr\n",
      "\n",
      "\n",
      "Attaching package: ‘magrittr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    subtract\n",
      "\n",
      "\n",
      "Loading required package: BSgenome\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: BiocIO\n",
      "\n",
      "Loading required package: rtracklayer\n",
      "\n",
      "\n",
      "Attaching package: ‘rtracklayer’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocIO’:\n",
      "\n",
      "    FileForFormat\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biostrings’:\n",
      "\n",
      "    collapse, intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:XVector’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Setting default genome to Mm10.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ArchR)\n",
    "library(parallel)\n",
    "library(BSgenome.Mmusculus.UCSC.mm10)\n",
    "library(ggpubr)\n",
    "library(dplyr)\n",
    "\n",
    "addArchRGenome(\"mm10\")\n",
    "\n",
    "set.seed(147)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa585f86",
   "metadata": {},
   "source": [
    "# D30 Dataset: Generate scATAC Peak Set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e50b570c-2ca5-48b6-b036-e990c4f2312d",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj <- loadArchRProject('/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ec58eb61-a479-4531-b159-ccff85cf64e8",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset'"
      ],
      "text/latex": [
       "'/data/peer/sotougl/Fuchs/inflammatory\\_memory/scATAC/intermediate\\_outputs3/projects/ArchR\\_D30\\_min\\_NA\\_max\\_600\\_scpeakset'"
      ],
      "text/markdown": [
       "'/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset'"
      ],
      "text/plain": [
       "[1] \"/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "proj@projectMetadata$outputDirectory = '/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset'\n",
    "proj@projectMetadata$outputDirectory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3640b433-c538-4995-9a52-1a1dc8a1eff5",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-1c694aea146e-Date-2025-01-04_Time-22-23-01.191764.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "C1 (1 of 4) : CellGroups N = 2\n",
      "\n",
      "C2 (2 of 4) : CellGroups N = 2\n",
      "\n",
      "C3 (3 of 4) : CellGroups N = 2\n",
      "\n",
      "C4 (4 of 4) : CellGroups N = 2\n",
      "\n",
      "2025-01-04 22:23:02.095163 : Creating Coverage Files!, 0.015 mins elapsed.\n",
      "\n",
      "2025-01-04 22:23:02.097163 : Batch Execution w/ safelapply!, 0.015 mins elapsed.\n",
      "\n",
      "2025-01-04 22:23:02.099604 : Group C1._.D30_PIMQ (1 of 8) : Creating Group Coverage File : C1._.D30_PIMQ.insertions.coverage.h5, 0.015 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:23:27.706756 : Group C1._.D30_Ctrl (2 of 8) : Creating Group Coverage File : C1._.D30_Ctrl.insertions.coverage.h5, 0.442 mins elapsed.\n",
      "\n",
      "Number of Cells = 406\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:23:48.477403 : Group C2._.D30_Ctrl (3 of 8) : Creating Group Coverage File : C2._.D30_Ctrl.insertions.coverage.h5, 0.788 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:24:16.405168 : Group C2._.D30_PIMQ (4 of 8) : Creating Group Coverage File : C2._.D30_PIMQ.insertions.coverage.h5, 1.254 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:24:46.391982 : Group C3._.D30_Ctrl (5 of 8) : Creating Group Coverage File : C3._.D30_Ctrl.insertions.coverage.h5, 1.753 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:25:07.975726 : Group C3._.D30_PIMQ (6 of 8) : Creating Group Coverage File : C3._.D30_PIMQ.insertions.coverage.h5, 2.113 mins elapsed.\n",
      "\n",
      "Number of Cells = 385\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:25:28.782507 : Group C4._.D30_PIMQ (7 of 8) : Creating Group Coverage File : C4._.D30_PIMQ.insertions.coverage.h5, 2.46 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:25:52.534601 : Group C4._.D30_Ctrl (8 of 8) : Creating Group Coverage File : C4._.D30_Ctrl.insertions.coverage.h5, 2.856 mins elapsed.\n",
      "\n",
      "Number of Cells = 75\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 22:26:11.746896 : Adding Kmer Bias to Coverage Files!, 3.176 mins elapsed.\n",
      "\n",
      "Kmer Bias chr1 (1 of 21)\n",
      "\n",
      "chr1 \n",
      "Coverage File chr1 (1 of 8)\n",
      "\n",
      "Coverage File chr1 (2 of 8)\n",
      "\n",
      "Coverage File chr1 (3 of 8)\n",
      "\n",
      "Coverage File chr1 (4 of 8)\n",
      "\n",
      "Coverage File chr1 (5 of 8)\n",
      "\n",
      "Coverage File chr1 (6 of 8)\n",
      "\n",
      "Coverage File chr1 (7 of 8)\n",
      "\n",
      "Coverage File chr1 (8 of 8)\n",
      "\n",
      "Kmer Bias chr10 (2 of 21)\n",
      "\n",
      "chr10 \n",
      "Coverage File chr10 (1 of 8)\n",
      "\n",
      "Coverage File chr10 (2 of 8)\n",
      "\n",
      "Coverage File chr10 (3 of 8)\n",
      "\n",
      "Coverage File chr10 (4 of 8)\n",
      "\n",
      "Coverage File chr10 (5 of 8)\n",
      "\n",
      "Coverage File chr10 (6 of 8)\n",
      "\n",
      "Coverage File chr10 (7 of 8)\n",
      "\n",
      "Coverage File chr10 (8 of 8)\n",
      "\n",
      "Kmer Bias chr11 (3 of 21)\n",
      "\n",
      "chr11 \n",
      "Coverage File chr11 (1 of 8)\n",
      "\n",
      "Coverage File chr11 (2 of 8)\n",
      "\n",
      "Coverage File chr11 (3 of 8)\n",
      "\n",
      "Coverage File chr11 (4 of 8)\n",
      "\n",
      "Coverage File chr11 (5 of 8)\n",
      "\n",
      "Coverage File chr11 (6 of 8)\n",
      "\n",
      "Coverage File chr11 (7 of 8)\n",
      "\n",
      "Coverage File chr11 (8 of 8)\n",
      "\n",
      "Kmer Bias chr12 (4 of 21)\n",
      "\n",
      "chr12 \n",
      "Coverage File chr12 (1 of 8)\n",
      "\n",
      "Coverage File chr12 (2 of 8)\n",
      "\n",
      "Coverage File chr12 (3 of 8)\n",
      "\n",
      "Coverage File chr12 (4 of 8)\n",
      "\n",
      "Coverage File chr12 (5 of 8)\n",
      "\n",
      "Coverage File chr12 (6 of 8)\n",
      "\n",
      "Coverage File chr12 (7 of 8)\n",
      "\n",
      "Coverage File chr12 (8 of 8)\n",
      "\n",
      "Kmer Bias chr13 (5 of 21)\n",
      "\n",
      "chr13 \n",
      "Coverage File chr13 (1 of 8)\n",
      "\n",
      "Coverage File chr13 (2 of 8)\n",
      "\n",
      "Coverage File chr13 (3 of 8)\n",
      "\n",
      "Coverage File chr13 (4 of 8)\n",
      "\n",
      "Coverage File chr13 (5 of 8)\n",
      "\n",
      "Coverage File chr13 (6 of 8)\n",
      "\n",
      "Coverage File chr13 (7 of 8)\n",
      "\n",
      "Coverage File chr13 (8 of 8)\n",
      "\n",
      "Kmer Bias chr14 (6 of 21)\n",
      "\n",
      "chr14 \n",
      "Coverage File chr14 (1 of 8)\n",
      "\n",
      "Coverage File chr14 (2 of 8)\n",
      "\n",
      "Coverage File chr14 (3 of 8)\n",
      "\n",
      "Coverage File chr14 (4 of 8)\n",
      "\n",
      "Coverage File chr14 (5 of 8)\n",
      "\n",
      "Coverage File chr14 (6 of 8)\n",
      "\n",
      "Coverage File chr14 (7 of 8)\n",
      "\n",
      "Coverage File chr14 (8 of 8)\n",
      "\n",
      "Kmer Bias chr15 (7 of 21)\n",
      "\n",
      "chr15 \n",
      "Coverage File chr15 (1 of 8)\n",
      "\n",
      "Coverage File chr15 (2 of 8)\n",
      "\n",
      "Coverage File chr15 (3 of 8)\n",
      "\n",
      "Coverage File chr15 (4 of 8)\n",
      "\n",
      "Coverage File chr15 (5 of 8)\n",
      "\n",
      "Coverage File chr15 (6 of 8)\n",
      "\n",
      "Coverage File chr15 (7 of 8)\n",
      "\n",
      "Coverage File chr15 (8 of 8)\n",
      "\n",
      "Kmer Bias chr16 (8 of 21)\n",
      "\n",
      "chr16 \n",
      "Coverage File chr16 (1 of 8)\n",
      "\n",
      "Coverage File chr16 (2 of 8)\n",
      "\n",
      "Coverage File chr16 (3 of 8)\n",
      "\n",
      "Coverage File chr16 (4 of 8)\n",
      "\n",
      "Coverage File chr16 (5 of 8)\n",
      "\n",
      "Coverage File chr16 (6 of 8)\n",
      "\n",
      "Coverage File chr16 (7 of 8)\n",
      "\n",
      "Coverage File chr16 (8 of 8)\n",
      "\n",
      "Kmer Bias chr17 (9 of 21)\n",
      "\n",
      "chr17 \n",
      "Coverage File chr17 (1 of 8)\n",
      "\n",
      "Coverage File chr17 (2 of 8)\n",
      "\n",
      "Coverage File chr17 (3 of 8)\n",
      "\n",
      "Coverage File chr17 (4 of 8)\n",
      "\n",
      "Coverage File chr17 (5 of 8)\n",
      "\n",
      "Coverage File chr17 (6 of 8)\n",
      "\n",
      "Coverage File chr17 (7 of 8)\n",
      "\n",
      "Coverage File chr17 (8 of 8)\n",
      "\n",
      "Kmer Bias chr18 (10 of 21)\n",
      "\n",
      "chr18 \n",
      "Coverage File chr18 (1 of 8)\n",
      "\n",
      "Coverage File chr18 (2 of 8)\n",
      "\n",
      "Coverage File chr18 (3 of 8)\n",
      "\n",
      "Coverage File chr18 (4 of 8)\n",
      "\n",
      "Coverage File chr18 (5 of 8)\n",
      "\n",
      "Coverage File chr18 (6 of 8)\n",
      "\n",
      "Coverage File chr18 (7 of 8)\n",
      "\n",
      "Coverage File chr18 (8 of 8)\n",
      "\n",
      "Kmer Bias chr19 (11 of 21)\n",
      "\n",
      "chr19 \n",
      "Coverage File chr19 (1 of 8)\n",
      "\n",
      "Coverage File chr19 (2 of 8)\n",
      "\n",
      "Coverage File chr19 (3 of 8)\n",
      "\n",
      "Coverage File chr19 (4 of 8)\n",
      "\n",
      "Coverage File chr19 (5 of 8)\n",
      "\n",
      "Coverage File chr19 (6 of 8)\n",
      "\n",
      "Coverage File chr19 (7 of 8)\n",
      "\n",
      "Coverage File chr19 (8 of 8)\n",
      "\n",
      "Kmer Bias chr2 (12 of 21)\n",
      "\n",
      "chr2 \n",
      "Coverage File chr2 (1 of 8)\n",
      "\n",
      "Coverage File chr2 (2 of 8)\n",
      "\n",
      "Coverage File chr2 (3 of 8)\n",
      "\n",
      "Coverage File chr2 (4 of 8)\n",
      "\n",
      "Coverage File chr2 (5 of 8)\n",
      "\n",
      "Coverage File chr2 (6 of 8)\n",
      "\n",
      "Coverage File chr2 (7 of 8)\n",
      "\n",
      "Coverage File chr2 (8 of 8)\n",
      "\n",
      "Kmer Bias chr3 (13 of 21)\n",
      "\n",
      "chr3 \n",
      "Coverage File chr3 (1 of 8)\n",
      "\n",
      "Coverage File chr3 (2 of 8)\n",
      "\n",
      "Coverage File chr3 (3 of 8)\n",
      "\n",
      "Coverage File chr3 (4 of 8)\n",
      "\n",
      "Coverage File chr3 (5 of 8)\n",
      "\n",
      "Coverage File chr3 (6 of 8)\n",
      "\n",
      "Coverage File chr3 (7 of 8)\n",
      "\n",
      "Coverage File chr3 (8 of 8)\n",
      "\n",
      "Kmer Bias chr4 (14 of 21)\n",
      "\n",
      "chr4 \n",
      "Coverage File chr4 (1 of 8)\n",
      "\n",
      "Coverage File chr4 (2 of 8)\n",
      "\n",
      "Coverage File chr4 (3 of 8)\n",
      "\n",
      "Coverage File chr4 (4 of 8)\n",
      "\n",
      "Coverage File chr4 (5 of 8)\n",
      "\n",
      "Coverage File chr4 (6 of 8)\n",
      "\n",
      "Coverage File chr4 (7 of 8)\n",
      "\n",
      "Coverage File chr4 (8 of 8)\n",
      "\n",
      "Kmer Bias chr5 (15 of 21)\n",
      "\n",
      "chr5 \n",
      "Coverage File chr5 (1 of 8)\n",
      "\n",
      "Coverage File chr5 (2 of 8)\n",
      "\n",
      "Coverage File chr5 (3 of 8)\n",
      "\n",
      "Coverage File chr5 (4 of 8)\n",
      "\n",
      "Coverage File chr5 (5 of 8)\n",
      "\n",
      "Coverage File chr5 (6 of 8)\n",
      "\n",
      "Coverage File chr5 (7 of 8)\n",
      "\n",
      "Coverage File chr5 (8 of 8)\n",
      "\n",
      "Kmer Bias chr6 (16 of 21)\n",
      "\n",
      "chr6 \n",
      "Coverage File chr6 (1 of 8)\n",
      "\n",
      "Coverage File chr6 (2 of 8)\n",
      "\n",
      "Coverage File chr6 (3 of 8)\n",
      "\n",
      "Coverage File chr6 (4 of 8)\n",
      "\n",
      "Coverage File chr6 (5 of 8)\n",
      "\n",
      "Coverage File chr6 (6 of 8)\n",
      "\n",
      "Coverage File chr6 (7 of 8)\n",
      "\n",
      "Coverage File chr6 (8 of 8)\n",
      "\n",
      "Kmer Bias chr7 (17 of 21)\n",
      "\n",
      "chr7 \n",
      "Coverage File chr7 (1 of 8)\n",
      "\n",
      "Coverage File chr7 (2 of 8)\n",
      "\n",
      "Coverage File chr7 (3 of 8)\n",
      "\n",
      "Coverage File chr7 (4 of 8)\n",
      "\n",
      "Coverage File chr7 (5 of 8)\n",
      "\n",
      "Coverage File chr7 (6 of 8)\n",
      "\n",
      "Coverage File chr7 (7 of 8)\n",
      "\n",
      "Coverage File chr7 (8 of 8)\n",
      "\n",
      "Kmer Bias chr8 (18 of 21)\n",
      "\n",
      "chr8 \n",
      "Coverage File chr8 (1 of 8)\n",
      "\n",
      "Coverage File chr8 (2 of 8)\n",
      "\n",
      "Coverage File chr8 (3 of 8)\n",
      "\n",
      "Coverage File chr8 (4 of 8)\n",
      "\n",
      "Coverage File chr8 (5 of 8)\n",
      "\n",
      "Coverage File chr8 (6 of 8)\n",
      "\n",
      "Coverage File chr8 (7 of 8)\n",
      "\n",
      "Coverage File chr8 (8 of 8)\n",
      "\n",
      "Kmer Bias chr9 (19 of 21)\n",
      "\n",
      "chr9 \n",
      "Coverage File chr9 (1 of 8)\n",
      "\n",
      "Coverage File chr9 (2 of 8)\n",
      "\n",
      "Coverage File chr9 (3 of 8)\n",
      "\n",
      "Coverage File chr9 (4 of 8)\n",
      "\n",
      "Coverage File chr9 (5 of 8)\n",
      "\n",
      "Coverage File chr9 (6 of 8)\n",
      "\n",
      "Coverage File chr9 (7 of 8)\n",
      "\n",
      "Coverage File chr9 (8 of 8)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Kmer Bias chrX (20 of 21)\n",
      "\n",
      "chrX \n",
      "Coverage File chrX (1 of 8)\n",
      "\n",
      "Coverage File chrX (2 of 8)\n",
      "\n",
      "Coverage File chrX (3 of 8)\n",
      "\n",
      "Coverage File chrX (4 of 8)\n",
      "\n",
      "Coverage File chrX (5 of 8)\n",
      "\n",
      "Coverage File chrX (6 of 8)\n",
      "\n",
      "Coverage File chrX (7 of 8)\n",
      "\n",
      "Coverage File chrX (8 of 8)\n",
      "\n",
      "Kmer Bias chrY (21 of 21)\n",
      "\n",
      "chrY \n",
      "Coverage File chrY (1 of 8)\n",
      "\n",
      "Coverage File chrY (2 of 8)\n",
      "\n",
      "Coverage File chrY (3 of 8)\n",
      "\n",
      "Coverage File chrY (4 of 8)\n",
      "\n",
      "Coverage File chrY (5 of 8)\n",
      "\n",
      "Coverage File chrY (6 of 8)\n",
      "\n",
      "Coverage File chrY (7 of 8)\n",
      "\n",
      "Coverage File chrY (8 of 8)\n",
      "\n",
      "Completed Kmer Bias Calculation\n",
      "\n",
      "Adding Kmer Bias (1 of 8)\n",
      "\n",
      "Adding Kmer Bias (2 of 8)\n",
      "\n",
      "Adding Kmer Bias (3 of 8)\n",
      "\n",
      "Adding Kmer Bias (4 of 8)\n",
      "\n",
      "Adding Kmer Bias (5 of 8)\n",
      "\n",
      "Adding Kmer Bias (6 of 8)\n",
      "\n",
      "Adding Kmer Bias (7 of 8)\n",
      "\n",
      "Adding Kmer Bias (8 of 8)\n",
      "\n",
      "2025-01-04 22:28:36.876069 : Finished Creation of Coverage Files!, 5.595 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addGroupCoverages-1c694aea146e-Date-2025-01-04_Time-22-23-01.191764.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Generate group-level coverage tracks (fragments capped at 600bp for NFR enrichment)\n",
    "proj <- addGroupCoverages(proj, maxFragmentLength=600, force = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "550d1b1b",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-1c6946579a317-Date-2025-01-04_Time-22-42-38.213509.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Calling Peaks with Macs2\n",
      "\n",
      "2025-01-04 22:42:38.386608 : Peak Calling Parameters!, 0.003 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Group nCells nCellsUsed nReplicates nMin nMax maxPeaks\n",
      "C1    C1   1309        906           2  406  500   150000\n",
      "C2    C2   1535       1000           2  500  500   150000\n",
      "C3    C3   4597        885           2  385  500   150000\n",
      "C4    C4   3287        575           2   75  500   150000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-01-04 22:42:38.397211 : Batching Peak Calls!, 0.003 mins elapsed.\n",
      "\n",
      "2025-01-04 22:42:38.406472 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2025-01-04 22:42:38.408838 : Group 1 of 8, Calling Peaks with MACS2!, 0 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C1._.D30_PIMQ-1 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C1._.D30_PIMQ-1.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:43:55.251992 : Group 2 of 8, Calling Peaks with MACS2!, 1.281 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C1._.D30_Ctrl-2 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C1._.D30_Ctrl-2.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:44:52.013534 : Group 3 of 8, Calling Peaks with MACS2!, 2.227 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C2._.D30_Ctrl-3 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C2._.D30_Ctrl-3.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:45:49.247374 : Group 4 of 8, Calling Peaks with MACS2!, 3.181 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C2._.D30_PIMQ-4 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C2._.D30_PIMQ-4.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:47:06.235547 : Group 5 of 8, Calling Peaks with MACS2!, 4.464 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C3._.D30_Ctrl-5 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C3._.D30_Ctrl-5.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:48:05.040208 : Group 6 of 8, Calling Peaks with MACS2!, 5.444 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C3._.D30_PIMQ-6 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C3._.D30_PIMQ-6.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:48:53.928645 : Group 7 of 8, Calling Peaks with MACS2!, 6.259 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C4._.D30_PIMQ-7 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C4._.D30_PIMQ-7.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:50:07.300253 : Group 8 of 8, Calling Peaks with MACS2!, 7.482 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C4._.D30_Ctrl-8 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds/C4._.D30_Ctrl-8.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 22:50:31.716775 : Identifying Reproducible Peaks!, 7.892 mins elapsed.\n",
      "\n",
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/C1-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/C2-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/C3-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_epdsc/PeakCalls/C4-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-01-04 22:50:46.619655 : Creating Union Peak Set!, 8.14 mins elapsed.\n",
      "\n",
      "Converged after 4 iterations!\n",
      "\n",
      "2025-01-04 22:50:49.392085 : Finished Creating Union Peak Set (112211)!, 8.186 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Call reproducible peaks across all groups using MACS3\n",
    "proj <- addReproduciblePeakSet(proj, pathToMacs2  = '/home/sotougl/miniconda3/envs/archr_env/bin/macs3', force = TRUE, plot = FALSE, threads = 1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "45305803-9d2c-4d9e-8e40-ca05b84ac4dd",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-1c694167108-Date-2025-01-04_Time-22-50-58.718306.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-04 22:50:58.82891 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:00.905815 : Adding D30_PIMQ to PeakMatrix for Chr (1 of 20)!, 0.007 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:07.689057 : Adding D30_PIMQ to PeakMatrix for Chr (2 of 20)!, 0.12 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:12.377504 : Adding D30_PIMQ to PeakMatrix for Chr (3 of 20)!, 0.198 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:16.454214 : Adding D30_PIMQ to PeakMatrix for Chr (4 of 20)!, 0.266 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:20.646034 : Adding D30_PIMQ to PeakMatrix for Chr (5 of 20)!, 0.336 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:24.605887 : Adding D30_PIMQ to PeakMatrix for Chr (6 of 20)!, 0.402 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:28.138868 : Adding D30_PIMQ to PeakMatrix for Chr (7 of 20)!, 0.461 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:32.400099 : Adding D30_PIMQ to PeakMatrix for Chr (8 of 20)!, 0.532 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:35.908583 : Adding D30_PIMQ to PeakMatrix for Chr (9 of 20)!, 0.59 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:39.750726 : Adding D30_PIMQ to PeakMatrix for Chr (10 of 20)!, 0.654 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:43.27287 : Adding D30_PIMQ to PeakMatrix for Chr (11 of 20)!, 0.713 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:47.858915 : Adding D30_PIMQ to PeakMatrix for Chr (12 of 20)!, 0.79 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:51.426349 : Adding D30_PIMQ to PeakMatrix for Chr (13 of 20)!, 0.849 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:54.593203 : Adding D30_PIMQ to PeakMatrix for Chr (14 of 20)!, 0.902 mins elapsed.\n",
      "\n",
      "2025-01-04 22:51:57.420201 : Adding D30_PIMQ to PeakMatrix for Chr (15 of 20)!, 0.949 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:00.620868 : Adding D30_PIMQ to PeakMatrix for Chr (16 of 20)!, 1.002 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:03.468091 : Adding D30_PIMQ to PeakMatrix for Chr (17 of 20)!, 1.05 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:06.731396 : Adding D30_PIMQ to PeakMatrix for Chr (18 of 20)!, 1.104 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:09.28542 : Adding D30_PIMQ to PeakMatrix for Chr (19 of 20)!, 1.147 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:11.899326 : Adding D30_PIMQ to PeakMatrix for Chr (20 of 20)!, 1.19 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:17.301124 : Adding D30_Ctrl to PeakMatrix for Chr (1 of 20)!, 0.006 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:20.974647 : Adding D30_Ctrl to PeakMatrix for Chr (2 of 20)!, 0.067 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:24.990052 : Adding D30_Ctrl to PeakMatrix for Chr (3 of 20)!, 0.134 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:28.040007 : Adding D30_Ctrl to PeakMatrix for Chr (4 of 20)!, 0.185 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:31.772523 : Adding D30_Ctrl to PeakMatrix for Chr (5 of 20)!, 0.247 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:35.103996 : Adding D30_Ctrl to PeakMatrix for Chr (6 of 20)!, 0.303 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:38.199032 : Adding D30_Ctrl to PeakMatrix for Chr (7 of 20)!, 0.354 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:41.700796 : Adding D30_Ctrl to PeakMatrix for Chr (8 of 20)!, 0.413 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:44.713646 : Adding D30_Ctrl to PeakMatrix for Chr (9 of 20)!, 0.463 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:47.818086 : Adding D30_Ctrl to PeakMatrix for Chr (10 of 20)!, 0.515 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:50.822304 : Adding D30_Ctrl to PeakMatrix for Chr (11 of 20)!, 0.565 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:55.075926 : Adding D30_Ctrl to PeakMatrix for Chr (12 of 20)!, 0.636 mins elapsed.\n",
      "\n",
      "2025-01-04 22:52:57.759701 : Adding D30_Ctrl to PeakMatrix for Chr (13 of 20)!, 0.68 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:00.546483 : Adding D30_Ctrl to PeakMatrix for Chr (14 of 20)!, 0.727 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:03.084378 : Adding D30_Ctrl to PeakMatrix for Chr (15 of 20)!, 0.769 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:05.868574 : Adding D30_Ctrl to PeakMatrix for Chr (16 of 20)!, 0.815 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:08.42092 : Adding D30_Ctrl to PeakMatrix for Chr (17 of 20)!, 0.858 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:11.254702 : Adding D30_Ctrl to PeakMatrix for Chr (18 of 20)!, 0.905 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:13.621866 : Adding D30_Ctrl to PeakMatrix for Chr (19 of 20)!, 0.945 mins elapsed.\n",
      "\n",
      "2025-01-04 22:53:15.870836 : Adding D30_Ctrl to PeakMatrix for Chr (20 of 20)!, 0.982 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-1c694167108-Date-2025-01-04_Time-22-50-58.718306.log\n",
      "\n",
      "Copying ArchRProject to new outputDirectory : /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 2)\n",
      "\n",
      "Copying Arrow Files (2 of 2)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n",
      "Copying Other Files (1 of 7): D30_Ctrl\n",
      "\n",
      "Copying Other Files (2 of 7): D30_PIMQ\n",
      "\n",
      "Copying Other Files (3 of 7): Embeddings\n",
      "\n",
      "Copying Other Files (4 of 7): GroupCoverages\n",
      "\n",
      "Copying Other Files (5 of 7): IterativeLSI\n",
      "\n",
      "Copying Other Files (6 of 7): IterativeLSI2\n",
      "\n",
      "Copying Other Files (7 of 7): PeakCalls\n",
      "\n",
      "Saving ArchRProject...\n",
      "\n",
      "Loading ArchRProject...\n",
      "\n",
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Generate single-cell peak accessibility matrix (NFR fragments, no ceiling for high counts)\n",
    "proj <- addPeakMatrix(proj, maxFragmentLength=600, ceiling=10^9)\n",
    "\n",
    "# Persist project with peak annotations\n",
    "proj <- saveArchRProject(ArchRProj = proj, outputDirectory = '/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b968847",
   "metadata": {},
   "source": [
    "# Y1 Dataset: Generate scATAC Peak Set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7cc16c2f-0f3b-4044-b865-65401cfe3ab4",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj <- loadArchRProject('/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3154582d",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-1c694389ad392-Date-2025-01-04_Time-23-35-08.621259.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "C1 (1 of 6) : CellGroups N = 2\n",
      "\n",
      "C2 (2 of 6) : CellGroups N = 2\n",
      "\n",
      "C3 (3 of 6) : CellGroups N = 2\n",
      "\n",
      "C4 (4 of 6) : CellGroups N = 2\n",
      "\n",
      "C5 (5 of 6) : CellGroups N = 2\n",
      "\n",
      "C6 (6 of 6) : CellGroups N = 2\n",
      "\n",
      "2025-01-04 23:35:09.236537 : Creating Coverage Files!, 0.01 mins elapsed.\n",
      "\n",
      "2025-01-04 23:35:09.238498 : Batch Execution w/ safelapply!, 0.01 mins elapsed.\n",
      "\n",
      "2025-01-04 23:35:09.241064 : Group C1._.Y1_Ctrl (1 of 12) : Creating Group Coverage File : C1._.Y1_Ctrl.insertions.coverage.h5, 0.01 mins elapsed.\n",
      "\n",
      "Number of Cells = 138\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:35:30.639309 : Group C1._.Y1_PIMQ (2 of 12) : Creating Group Coverage File : C1._.Y1_PIMQ.insertions.coverage.h5, 0.367 mins elapsed.\n",
      "\n",
      "Number of Cells = 119\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:35:50.697748 : Group C2._.Y1_Ctrl (3 of 12) : Creating Group Coverage File : C2._.Y1_Ctrl.insertions.coverage.h5, 0.701 mins elapsed.\n",
      "\n",
      "Number of Cells = 226\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:36:15.474855 : Group C2._.Y1_PIMQ (4 of 12) : Creating Group Coverage File : C2._.Y1_PIMQ.insertions.coverage.h5, 1.114 mins elapsed.\n",
      "\n",
      "Number of Cells = 116\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:36:35.885898 : Group C3._.Y1_Ctrl (5 of 12) : Creating Group Coverage File : C3._.Y1_Ctrl.insertions.coverage.h5, 1.454 mins elapsed.\n",
      "\n",
      "Number of Cells = 262\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:37:02.290978 : Group C3._.Y1_PIMQ (6 of 12) : Creating Group Coverage File : C3._.Y1_PIMQ.insertions.coverage.h5, 1.894 mins elapsed.\n",
      "\n",
      "Number of Cells = 226\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:37:26.124314 : Group C4._.Y1_PIMQ (7 of 12) : Creating Group Coverage File : C4._.Y1_PIMQ.insertions.coverage.h5, 2.292 mins elapsed.\n",
      "\n",
      "Number of Cells = 148\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:37:48.139236 : Group C4._.Y1_Ctrl (8 of 12) : Creating Group Coverage File : C4._.Y1_Ctrl.insertions.coverage.h5, 2.659 mins elapsed.\n",
      "\n",
      "Number of Cells = 99\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:38:08.864371 : Group C5._.Y1_PIMQ (9 of 12) : Creating Group Coverage File : C5._.Y1_PIMQ.insertions.coverage.h5, 3.004 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:38:42.100904 : Group C5._.Y1_Ctrl (10 of 12) : Creating Group Coverage File : C5._.Y1_Ctrl.insertions.coverage.h5, 3.558 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:39:16.879867 : Group C6._.Y1_Ctrl (11 of 12) : Creating Group Coverage File : C6._.Y1_Ctrl.insertions.coverage.h5, 4.138 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:39:47.560826 : Group C6._.Y1_PIMQ (12 of 12) : Creating Group Coverage File : C6._.Y1_PIMQ.insertions.coverage.h5, 4.649 mins elapsed.\n",
      "\n",
      "Number of Cells = 428\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2025-01-04 23:40:15.477792 : Adding Kmer Bias to Coverage Files!, 5.114 mins elapsed.\n",
      "\n",
      "Kmer Bias chr1 (1 of 21)\n",
      "\n",
      "chr1 \n",
      "Coverage File chr1 (1 of 12)\n",
      "\n",
      "Coverage File chr1 (2 of 12)\n",
      "\n",
      "Coverage File chr1 (3 of 12)\n",
      "\n",
      "Coverage File chr1 (4 of 12)\n",
      "\n",
      "Coverage File chr1 (5 of 12)\n",
      "\n",
      "Coverage File chr1 (6 of 12)\n",
      "\n",
      "Coverage File chr1 (7 of 12)\n",
      "\n",
      "Coverage File chr1 (8 of 12)\n",
      "\n",
      "Coverage File chr1 (9 of 12)\n",
      "\n",
      "Coverage File chr1 (10 of 12)\n",
      "\n",
      "Coverage File chr1 (11 of 12)\n",
      "\n",
      "Coverage File chr1 (12 of 12)\n",
      "\n",
      "Kmer Bias chr10 (2 of 21)\n",
      "\n",
      "chr10 \n",
      "Coverage File chr10 (1 of 12)\n",
      "\n",
      "Coverage File chr10 (2 of 12)\n",
      "\n",
      "Coverage File chr10 (3 of 12)\n",
      "\n",
      "Coverage File chr10 (4 of 12)\n",
      "\n",
      "Coverage File chr10 (5 of 12)\n",
      "\n",
      "Coverage File chr10 (6 of 12)\n",
      "\n",
      "Coverage File chr10 (7 of 12)\n",
      "\n",
      "Coverage File chr10 (8 of 12)\n",
      "\n",
      "Coverage File chr10 (9 of 12)\n",
      "\n",
      "Coverage File chr10 (10 of 12)\n",
      "\n",
      "Coverage File chr10 (11 of 12)\n",
      "\n",
      "Coverage File chr10 (12 of 12)\n",
      "\n",
      "Kmer Bias chr11 (3 of 21)\n",
      "\n",
      "chr11 \n",
      "Coverage File chr11 (1 of 12)\n",
      "\n",
      "Coverage File chr11 (2 of 12)\n",
      "\n",
      "Coverage File chr11 (3 of 12)\n",
      "\n",
      "Coverage File chr11 (4 of 12)\n",
      "\n",
      "Coverage File chr11 (5 of 12)\n",
      "\n",
      "Coverage File chr11 (6 of 12)\n",
      "\n",
      "Coverage File chr11 (7 of 12)\n",
      "\n",
      "Coverage File chr11 (8 of 12)\n",
      "\n",
      "Coverage File chr11 (9 of 12)\n",
      "\n",
      "Coverage File chr11 (10 of 12)\n",
      "\n",
      "Coverage File chr11 (11 of 12)\n",
      "\n",
      "Coverage File chr11 (12 of 12)\n",
      "\n",
      "Kmer Bias chr12 (4 of 21)\n",
      "\n",
      "chr12 \n",
      "Coverage File chr12 (1 of 12)\n",
      "\n",
      "Coverage File chr12 (2 of 12)\n",
      "\n",
      "Coverage File chr12 (3 of 12)\n",
      "\n",
      "Coverage File chr12 (4 of 12)\n",
      "\n",
      "Coverage File chr12 (5 of 12)\n",
      "\n",
      "Coverage File chr12 (6 of 12)\n",
      "\n",
      "Coverage File chr12 (7 of 12)\n",
      "\n",
      "Coverage File chr12 (8 of 12)\n",
      "\n",
      "Coverage File chr12 (9 of 12)\n",
      "\n",
      "Coverage File chr12 (10 of 12)\n",
      "\n",
      "Coverage File chr12 (11 of 12)\n",
      "\n",
      "Coverage File chr12 (12 of 12)\n",
      "\n",
      "Kmer Bias chr13 (5 of 21)\n",
      "\n",
      "chr13 \n",
      "Coverage File chr13 (1 of 12)\n",
      "\n",
      "Coverage File chr13 (2 of 12)\n",
      "\n",
      "Coverage File chr13 (3 of 12)\n",
      "\n",
      "Coverage File chr13 (4 of 12)\n",
      "\n",
      "Coverage File chr13 (5 of 12)\n",
      "\n",
      "Coverage File chr13 (6 of 12)\n",
      "\n",
      "Coverage File chr13 (7 of 12)\n",
      "\n",
      "Coverage File chr13 (8 of 12)\n",
      "\n",
      "Coverage File chr13 (9 of 12)\n",
      "\n",
      "Coverage File chr13 (10 of 12)\n",
      "\n",
      "Coverage File chr13 (11 of 12)\n",
      "\n",
      "Coverage File chr13 (12 of 12)\n",
      "\n",
      "Kmer Bias chr14 (6 of 21)\n",
      "\n",
      "chr14 \n",
      "Coverage File chr14 (1 of 12)\n",
      "\n",
      "Coverage File chr14 (2 of 12)\n",
      "\n",
      "Coverage File chr14 (3 of 12)\n",
      "\n",
      "Coverage File chr14 (4 of 12)\n",
      "\n",
      "Coverage File chr14 (5 of 12)\n",
      "\n",
      "Coverage File chr14 (6 of 12)\n",
      "\n",
      "Coverage File chr14 (7 of 12)\n",
      "\n",
      "Coverage File chr14 (8 of 12)\n",
      "\n",
      "Coverage File chr14 (9 of 12)\n",
      "\n",
      "Coverage File chr14 (10 of 12)\n",
      "\n",
      "Coverage File chr14 (11 of 12)\n",
      "\n",
      "Coverage File chr14 (12 of 12)\n",
      "\n",
      "Kmer Bias chr15 (7 of 21)\n",
      "\n",
      "chr15 \n",
      "Coverage File chr15 (1 of 12)\n",
      "\n",
      "Coverage File chr15 (2 of 12)\n",
      "\n",
      "Coverage File chr15 (3 of 12)\n",
      "\n",
      "Coverage File chr15 (4 of 12)\n",
      "\n",
      "Coverage File chr15 (5 of 12)\n",
      "\n",
      "Coverage File chr15 (6 of 12)\n",
      "\n",
      "Coverage File chr15 (7 of 12)\n",
      "\n",
      "Coverage File chr15 (8 of 12)\n",
      "\n",
      "Coverage File chr15 (9 of 12)\n",
      "\n",
      "Coverage File chr15 (10 of 12)\n",
      "\n",
      "Coverage File chr15 (11 of 12)\n",
      "\n",
      "Coverage File chr15 (12 of 12)\n",
      "\n",
      "Kmer Bias chr16 (8 of 21)\n",
      "\n",
      "chr16 \n",
      "Coverage File chr16 (1 of 12)\n",
      "\n",
      "Coverage File chr16 (2 of 12)\n",
      "\n",
      "Coverage File chr16 (3 of 12)\n",
      "\n",
      "Coverage File chr16 (4 of 12)\n",
      "\n",
      "Coverage File chr16 (5 of 12)\n",
      "\n",
      "Coverage File chr16 (6 of 12)\n",
      "\n",
      "Coverage File chr16 (7 of 12)\n",
      "\n",
      "Coverage File chr16 (8 of 12)\n",
      "\n",
      "Coverage File chr16 (9 of 12)\n",
      "\n",
      "Coverage File chr16 (10 of 12)\n",
      "\n",
      "Coverage File chr16 (11 of 12)\n",
      "\n",
      "Coverage File chr16 (12 of 12)\n",
      "\n",
      "Kmer Bias chr17 (9 of 21)\n",
      "\n",
      "chr17 \n",
      "Coverage File chr17 (1 of 12)\n",
      "\n",
      "Coverage File chr17 (2 of 12)\n",
      "\n",
      "Coverage File chr17 (3 of 12)\n",
      "\n",
      "Coverage File chr17 (4 of 12)\n",
      "\n",
      "Coverage File chr17 (5 of 12)\n",
      "\n",
      "Coverage File chr17 (6 of 12)\n",
      "\n",
      "Coverage File chr17 (7 of 12)\n",
      "\n",
      "Coverage File chr17 (8 of 12)\n",
      "\n",
      "Coverage File chr17 (9 of 12)\n",
      "\n",
      "Coverage File chr17 (10 of 12)\n",
      "\n",
      "Coverage File chr17 (11 of 12)\n",
      "\n",
      "Coverage File chr17 (12 of 12)\n",
      "\n",
      "Kmer Bias chr18 (10 of 21)\n",
      "\n",
      "chr18 \n",
      "Coverage File chr18 (1 of 12)\n",
      "\n",
      "Coverage File chr18 (2 of 12)\n",
      "\n",
      "Coverage File chr18 (3 of 12)\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Coverage File chr18 (4 of 12)\n",
      "\n",
      "Coverage File chr18 (5 of 12)\n",
      "\n",
      "Coverage File chr18 (6 of 12)\n",
      "\n",
      "Coverage File chr18 (7 of 12)\n",
      "\n",
      "Coverage File chr18 (8 of 12)\n",
      "\n",
      "Coverage File chr18 (9 of 12)\n",
      "\n",
      "Coverage File chr18 (10 of 12)\n",
      "\n",
      "Coverage File chr18 (11 of 12)\n",
      "\n",
      "Coverage File chr18 (12 of 12)\n",
      "\n",
      "Kmer Bias chr19 (11 of 21)\n",
      "\n",
      "chr19 \n",
      "Coverage File chr19 (1 of 12)\n",
      "\n",
      "Coverage File chr19 (2 of 12)\n",
      "\n",
      "Coverage File chr19 (3 of 12)\n",
      "\n",
      "Coverage File chr19 (4 of 12)\n",
      "\n",
      "Coverage File chr19 (5 of 12)\n",
      "\n",
      "Coverage File chr19 (6 of 12)\n",
      "\n",
      "Coverage File chr19 (7 of 12)\n",
      "\n",
      "Coverage File chr19 (8 of 12)\n",
      "\n",
      "Coverage File chr19 (9 of 12)\n",
      "\n",
      "Coverage File chr19 (10 of 12)\n",
      "\n",
      "Coverage File chr19 (11 of 12)\n",
      "\n",
      "Coverage File chr19 (12 of 12)\n",
      "\n",
      "Kmer Bias chr2 (12 of 21)\n",
      "\n",
      "chr2 \n",
      "Coverage File chr2 (1 of 12)\n",
      "\n",
      "Coverage File chr2 (2 of 12)\n",
      "\n",
      "Coverage File chr2 (3 of 12)\n",
      "\n",
      "Coverage File chr2 (4 of 12)\n",
      "\n",
      "Coverage File chr2 (5 of 12)\n",
      "\n",
      "Coverage File chr2 (6 of 12)\n",
      "\n",
      "Coverage File chr2 (7 of 12)\n",
      "\n",
      "Coverage File chr2 (8 of 12)\n",
      "\n",
      "Coverage File chr2 (9 of 12)\n",
      "\n",
      "Coverage File chr2 (10 of 12)\n",
      "\n",
      "Coverage File chr2 (11 of 12)\n",
      "\n",
      "Coverage File chr2 (12 of 12)\n",
      "\n",
      "Kmer Bias chr3 (13 of 21)\n",
      "\n",
      "chr3 \n",
      "Coverage File chr3 (1 of 12)\n",
      "\n",
      "Coverage File chr3 (2 of 12)\n",
      "\n",
      "Coverage File chr3 (3 of 12)\n",
      "\n",
      "Coverage File chr3 (4 of 12)\n",
      "\n",
      "Coverage File chr3 (5 of 12)\n",
      "\n",
      "Coverage File chr3 (6 of 12)\n",
      "\n",
      "Coverage File chr3 (7 of 12)\n",
      "\n",
      "Coverage File chr3 (8 of 12)\n",
      "\n",
      "Coverage File chr3 (9 of 12)\n",
      "\n",
      "Coverage File chr3 (10 of 12)\n",
      "\n",
      "Coverage File chr3 (11 of 12)\n",
      "\n",
      "Coverage File chr3 (12 of 12)\n",
      "\n",
      "Kmer Bias chr4 (14 of 21)\n",
      "\n",
      "chr4 \n",
      "Coverage File chr4 (1 of 12)\n",
      "\n",
      "Coverage File chr4 (2 of 12)\n",
      "\n",
      "Coverage File chr4 (3 of 12)\n",
      "\n",
      "Coverage File chr4 (4 of 12)\n",
      "\n",
      "Coverage File chr4 (5 of 12)\n",
      "\n",
      "Coverage File chr4 (6 of 12)\n",
      "\n",
      "Coverage File chr4 (7 of 12)\n",
      "\n",
      "Coverage File chr4 (8 of 12)\n",
      "\n",
      "Coverage File chr4 (9 of 12)\n",
      "\n",
      "Coverage File chr4 (10 of 12)\n",
      "\n",
      "Coverage File chr4 (11 of 12)\n",
      "\n",
      "Coverage File chr4 (12 of 12)\n",
      "\n",
      "Kmer Bias chr5 (15 of 21)\n",
      "\n",
      "chr5 \n",
      "Coverage File chr5 (1 of 12)\n",
      "\n",
      "Coverage File chr5 (2 of 12)\n",
      "\n",
      "Coverage File chr5 (3 of 12)\n",
      "\n",
      "Coverage File chr5 (4 of 12)\n",
      "\n",
      "Coverage File chr5 (5 of 12)\n",
      "\n",
      "Coverage File chr5 (6 of 12)\n",
      "\n",
      "Coverage File chr5 (7 of 12)\n",
      "\n",
      "Coverage File chr5 (8 of 12)\n",
      "\n",
      "Coverage File chr5 (9 of 12)\n",
      "\n",
      "Coverage File chr5 (10 of 12)\n",
      "\n",
      "Coverage File chr5 (11 of 12)\n",
      "\n",
      "Coverage File chr5 (12 of 12)\n",
      "\n",
      "Kmer Bias chr6 (16 of 21)\n",
      "\n",
      "chr6 \n",
      "Coverage File chr6 (1 of 12)\n",
      "\n",
      "Coverage File chr6 (2 of 12)\n",
      "\n",
      "Coverage File chr6 (3 of 12)\n",
      "\n",
      "Coverage File chr6 (4 of 12)\n",
      "\n",
      "Coverage File chr6 (5 of 12)\n",
      "\n",
      "Coverage File chr6 (6 of 12)\n",
      "\n",
      "Coverage File chr6 (7 of 12)\n",
      "\n",
      "Coverage File chr6 (8 of 12)\n",
      "\n",
      "Coverage File chr6 (9 of 12)\n",
      "\n",
      "Coverage File chr6 (10 of 12)\n",
      "\n",
      "Coverage File chr6 (11 of 12)\n",
      "\n",
      "Coverage File chr6 (12 of 12)\n",
      "\n",
      "Kmer Bias chr7 (17 of 21)\n",
      "\n",
      "chr7 \n",
      "Coverage File chr7 (1 of 12)\n",
      "\n",
      "Coverage File chr7 (2 of 12)\n",
      "\n",
      "Coverage File chr7 (3 of 12)\n",
      "\n",
      "Coverage File chr7 (4 of 12)\n",
      "\n",
      "Coverage File chr7 (5 of 12)\n",
      "\n",
      "Coverage File chr7 (6 of 12)\n",
      "\n",
      "Coverage File chr7 (7 of 12)\n",
      "\n",
      "Coverage File chr7 (8 of 12)\n",
      "\n",
      "Coverage File chr7 (9 of 12)\n",
      "\n",
      "Coverage File chr7 (10 of 12)\n",
      "\n",
      "Coverage File chr7 (11 of 12)\n",
      "\n",
      "Coverage File chr7 (12 of 12)\n",
      "\n",
      "Kmer Bias chr8 (18 of 21)\n",
      "\n",
      "chr8 \n",
      "Coverage File chr8 (1 of 12)\n",
      "\n",
      "Coverage File chr8 (2 of 12)\n",
      "\n",
      "Coverage File chr8 (3 of 12)\n",
      "\n",
      "Coverage File chr8 (4 of 12)\n",
      "\n",
      "Coverage File chr8 (5 of 12)\n",
      "\n",
      "Coverage File chr8 (6 of 12)\n",
      "\n",
      "Coverage File chr8 (7 of 12)\n",
      "\n",
      "Coverage File chr8 (8 of 12)\n",
      "\n",
      "Coverage File chr8 (9 of 12)\n",
      "\n",
      "Coverage File chr8 (10 of 12)\n",
      "\n",
      "Coverage File chr8 (11 of 12)\n",
      "\n",
      "Coverage File chr8 (12 of 12)\n",
      "\n",
      "Kmer Bias chr9 (19 of 21)\n",
      "\n",
      "chr9 \n",
      "Coverage File chr9 (1 of 12)\n",
      "\n",
      "Coverage File chr9 (2 of 12)\n",
      "\n",
      "Coverage File chr9 (3 of 12)\n",
      "\n",
      "Coverage File chr9 (4 of 12)\n",
      "\n",
      "Coverage File chr9 (5 of 12)\n",
      "\n",
      "Coverage File chr9 (6 of 12)\n",
      "\n",
      "Coverage File chr9 (7 of 12)\n",
      "\n",
      "Coverage File chr9 (8 of 12)\n",
      "\n",
      "Coverage File chr9 (9 of 12)\n",
      "\n",
      "Coverage File chr9 (10 of 12)\n",
      "\n",
      "Coverage File chr9 (11 of 12)\n",
      "\n",
      "Coverage File chr9 (12 of 12)\n",
      "\n",
      "Kmer Bias chrX (20 of 21)\n",
      "\n",
      "chrX \n",
      "Coverage File chrX (1 of 12)\n",
      "\n",
      "Coverage File chrX (2 of 12)\n",
      "\n",
      "Coverage File chrX (3 of 12)\n",
      "\n",
      "Coverage File chrX (4 of 12)\n",
      "\n",
      "Coverage File chrX (5 of 12)\n",
      "\n",
      "Coverage File chrX (6 of 12)\n",
      "\n",
      "Coverage File chrX (7 of 12)\n",
      "\n",
      "Coverage File chrX (8 of 12)\n",
      "\n",
      "Coverage File chrX (9 of 12)\n",
      "\n",
      "Coverage File chrX (10 of 12)\n",
      "\n",
      "Coverage File chrX (11 of 12)\n",
      "\n",
      "Coverage File chrX (12 of 12)\n",
      "\n",
      "Kmer Bias chrY (21 of 21)\n",
      "\n",
      "chrY \n",
      "Coverage File chrY (1 of 12)\n",
      "\n",
      "Coverage File chrY (2 of 12)\n",
      "\n",
      "Coverage File chrY (3 of 12)\n",
      "\n",
      "Coverage File chrY (4 of 12)\n",
      "\n",
      "Coverage File chrY (5 of 12)\n",
      "\n",
      "Coverage File chrY (6 of 12)\n",
      "\n",
      "Coverage File chrY (7 of 12)\n",
      "\n",
      "Coverage File chrY (8 of 12)\n",
      "\n",
      "Coverage File chrY (9 of 12)\n",
      "\n",
      "Coverage File chrY (10 of 12)\n",
      "\n",
      "Coverage File chrY (11 of 12)\n",
      "\n",
      "Coverage File chrY (12 of 12)\n",
      "\n",
      "Completed Kmer Bias Calculation\n",
      "\n",
      "Adding Kmer Bias (1 of 12)\n",
      "\n",
      "Adding Kmer Bias (2 of 12)\n",
      "\n",
      "Adding Kmer Bias (3 of 12)\n",
      "\n",
      "Adding Kmer Bias (4 of 12)\n",
      "\n",
      "Adding Kmer Bias (5 of 12)\n",
      "\n",
      "Adding Kmer Bias (6 of 12)\n",
      "\n",
      "Adding Kmer Bias (7 of 12)\n",
      "\n",
      "Adding Kmer Bias (8 of 12)\n",
      "\n",
      "Adding Kmer Bias (9 of 12)\n",
      "\n",
      "Adding Kmer Bias (10 of 12)\n",
      "\n",
      "Adding Kmer Bias (11 of 12)\n",
      "\n",
      "Adding Kmer Bias (12 of 12)\n",
      "\n",
      "2025-01-04 23:44:35.179626 : Finished Creation of Coverage Files!, 9.443 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addGroupCoverages-1c694389ad392-Date-2025-01-04_Time-23-35-08.621259.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Generate group-level coverage tracks (fragments capped at 600bp for NFR enrichment)\n",
    "proj <- addGroupCoverages(proj, maxFragmentLength=600, force = TRUE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60457832-7eab-4ece-a816-b89974a0bade",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-1c69475e619ea-Date-2025-01-04_Time-23-44-35.554113.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Calling Peaks with Macs2\n",
      "\n",
      "2025-01-04 23:44:35.705258 : Peak Calling Parameters!, 0.003 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   Group nCells nCellsUsed nReplicates nMin nMax maxPeaks\n",
      "C1    C1    257        257           2  119  138   128500\n",
      "C2    C2    342        342           2  116  226   150000\n",
      "C3    C3    488        488           2  226  262   150000\n",
      "C4    C4    247        247           2   99  148   123500\n",
      "C5    C5   2095       1000           2  500  500   150000\n",
      "C6    C6   1047        928           2  428  500   150000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-01-04 23:44:35.71435 : Batching Peak Calls!, 0.003 mins elapsed.\n",
      "\n",
      "2025-01-04 23:44:35.722327 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2025-01-04 23:44:35.724512 : Group 1 of 12, Calling Peaks with MACS2!, 0 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C1._.Y1_Ctrl-1 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C1._.Y1_Ctrl-1.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:45:51.409316 : Group 2 of 12, Calling Peaks with MACS2!, 1.261 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C1._.Y1_PIMQ-2 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C1._.Y1_PIMQ-2.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:46:55.317765 : Group 3 of 12, Calling Peaks with MACS2!, 2.327 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C2._.Y1_Ctrl-3 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C2._.Y1_Ctrl-3.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:48:38.514007 : Group 4 of 12, Calling Peaks with MACS2!, 4.047 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C2._.Y1_PIMQ-4 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C2._.Y1_PIMQ-4.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:49:46.290161 : Group 5 of 12, Calling Peaks with MACS2!, 5.176 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C3._.Y1_Ctrl-5 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C3._.Y1_Ctrl-5.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:51:34.632818 : Group 6 of 12, Calling Peaks with MACS2!, 6.982 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C3._.Y1_PIMQ-6 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C3._.Y1_PIMQ-6.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:53:09.476635 : Group 7 of 12, Calling Peaks with MACS2!, 8.563 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C4._.Y1_PIMQ-7 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C4._.Y1_PIMQ-7.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:54:27.446815 : Group 8 of 12, Calling Peaks with MACS2!, 9.862 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C4._.Y1_Ctrl-8 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C4._.Y1_Ctrl-8.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:55:30.063553 : Group 9 of 12, Calling Peaks with MACS2!, 10.906 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C5._.Y1_PIMQ-9 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C5._.Y1_PIMQ-9.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-04 23:58:06.759444 : Group 10 of 12, Calling Peaks with MACS2!, 13.517 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C5._.Y1_Ctrl-10 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C5._.Y1_Ctrl-10.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-05 00:00:58.374368 : Group 11 of 12, Calling Peaks with MACS2!, 16.378 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C6._.Y1_Ctrl-11 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C6._.Y1_Ctrl-11.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-05 00:03:10.96134 : Group 12 of 12, Calling Peaks with MACS2!, 18.587 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 1.87e+09 --name C6._.Y1_PIMQ-12 --treatment /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds/C6._.Y1_PIMQ-12.insertions.bed --outdir /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2025-01-05 00:05:11.6606 : Identifying Reproducible Peaks!, 20.602 mins elapsed.\n",
      "\n",
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/C1-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/C2-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/C3-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/C4-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/C5-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_epdsc/PeakCalls/C6-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-01-05 00:05:39.946452 : Creating Union Peak Set!, 21.073 mins elapsed.\n",
      "\n",
      "Converged after 7 iterations!\n",
      "\n",
      "2025-01-05 00:05:44.560717 : Finished Creating Union Peak Set (164945)!, 21.15 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-1c694c269340-Date-2025-01-05_Time-00-05-44.56543.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 00:05:44.670176 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2025-01-05 00:05:45.375527 : Adding Y1_PIMQ to PeakMatrix for Chr (1 of 20)!, 0.008 mins elapsed.\n",
      "\n",
      "2025-01-05 00:05:51.585205 : Adding Y1_PIMQ to PeakMatrix for Chr (2 of 20)!, 0.112 mins elapsed.\n",
      "\n",
      "2025-01-05 00:05:58.292212 : Adding Y1_PIMQ to PeakMatrix for Chr (3 of 20)!, 0.223 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:02.440451 : Adding Y1_PIMQ to PeakMatrix for Chr (4 of 20)!, 0.292 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:07.607108 : Adding Y1_PIMQ to PeakMatrix for Chr (5 of 20)!, 0.379 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:12.931926 : Adding Y1_PIMQ to PeakMatrix for Chr (6 of 20)!, 0.467 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:17.025468 : Adding Y1_PIMQ to PeakMatrix for Chr (7 of 20)!, 0.536 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:22.255368 : Adding Y1_PIMQ to PeakMatrix for Chr (8 of 20)!, 0.623 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:27.082968 : Adding Y1_PIMQ to PeakMatrix for Chr (9 of 20)!, 0.703 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:31.596699 : Adding Y1_PIMQ to PeakMatrix for Chr (10 of 20)!, 0.778 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:35.698898 : Adding Y1_PIMQ to PeakMatrix for Chr (11 of 20)!, 0.847 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:42.501304 : Adding Y1_PIMQ to PeakMatrix for Chr (12 of 20)!, 0.96 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:46.040396 : Adding Y1_PIMQ to PeakMatrix for Chr (13 of 20)!, 1.019 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:49.789722 : Adding Y1_PIMQ to PeakMatrix for Chr (14 of 20)!, 1.082 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:53.045519 : Adding Y1_PIMQ to PeakMatrix for Chr (15 of 20)!, 1.136 mins elapsed.\n",
      "\n",
      "2025-01-05 00:06:56.821872 : Adding Y1_PIMQ to PeakMatrix for Chr (16 of 20)!, 1.199 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:00.105413 : Adding Y1_PIMQ to PeakMatrix for Chr (17 of 20)!, 1.254 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:04.050707 : Adding Y1_PIMQ to PeakMatrix for Chr (18 of 20)!, 1.319 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:06.984782 : Adding Y1_PIMQ to PeakMatrix for Chr (19 of 20)!, 1.368 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:10.084169 : Adding Y1_PIMQ to PeakMatrix for Chr (20 of 20)!, 1.42 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:13.202111 : Adding Y1_Ctrl to PeakMatrix for Chr (1 of 20)!, 0.008 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:18.785119 : Adding Y1_Ctrl to PeakMatrix for Chr (2 of 20)!, 0.101 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:26.199508 : Adding Y1_Ctrl to PeakMatrix for Chr (3 of 20)!, 0.225 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:30.43885 : Adding Y1_Ctrl to PeakMatrix for Chr (4 of 20)!, 0.295 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:36.249276 : Adding Y1_Ctrl to PeakMatrix for Chr (5 of 20)!, 0.392 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:41.322913 : Adding Y1_Ctrl to PeakMatrix for Chr (6 of 20)!, 0.477 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:45.977362 : Adding Y1_Ctrl to PeakMatrix for Chr (7 of 20)!, 0.554 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:51.829544 : Adding Y1_Ctrl to PeakMatrix for Chr (8 of 20)!, 0.652 mins elapsed.\n",
      "\n",
      "2025-01-05 00:07:57.063844 : Adding Y1_Ctrl to PeakMatrix for Chr (9 of 20)!, 0.739 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:01.666412 : Adding Y1_Ctrl to PeakMatrix for Chr (10 of 20)!, 0.816 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:05.835191 : Adding Y1_Ctrl to PeakMatrix for Chr (11 of 20)!, 0.885 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:12.32837 : Adding Y1_Ctrl to PeakMatrix for Chr (12 of 20)!, 0.993 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:15.971859 : Adding Y1_Ctrl to PeakMatrix for Chr (13 of 20)!, 1.054 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:19.725253 : Adding Y1_Ctrl to PeakMatrix for Chr (14 of 20)!, 1.117 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:22.996803 : Adding Y1_Ctrl to PeakMatrix for Chr (15 of 20)!, 1.171 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:26.752918 : Adding Y1_Ctrl to PeakMatrix for Chr (16 of 20)!, 1.234 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:30.063955 : Adding Y1_Ctrl to PeakMatrix for Chr (17 of 20)!, 1.289 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:34.194108 : Adding Y1_Ctrl to PeakMatrix for Chr (18 of 20)!, 1.358 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:37.144012 : Adding Y1_Ctrl to PeakMatrix for Chr (19 of 20)!, 1.407 mins elapsed.\n",
      "\n",
      "2025-01-05 00:08:40.264356 : Adding Y1_Ctrl to PeakMatrix for Chr (20 of 20)!, 1.459 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-1c694c269340-Date-2025-01-05_Time-00-05-44.56543.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Call reproducible peaks across all groups using MACS3\n",
    "proj <- addReproduciblePeakSet(proj, pathToMacs2  = '/home/sotougl/miniconda3/envs/archr_env/bin/macs3', force = TRUE, plot = FALSE, threads = 1)\n",
    "\n",
    "# Generate single-cell peak accessibility matrix (NFR fragments, no ceiling for high counts)\n",
    "proj <- addPeakMatrix(proj, maxFragmentLength=600, ceiling=10^9)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fcf5b7d5-e591-4107-b294-8736df6917f0",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Copying ArchRProject to new outputDirectory : /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_scpeakset\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 2)\n",
      "\n",
      "Copying Arrow Files (2 of 2)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n",
      "Copying Other Files (1 of 7): Embeddings\n",
      "\n",
      "Copying Other Files (2 of 7): GroupCoverages\n",
      "\n",
      "Copying Other Files (3 of 7): IterativeLSI\n",
      "\n",
      "Copying Other Files (4 of 7): IterativeLSI2\n",
      "\n",
      "Copying Other Files (5 of 7): PeakCalls\n",
      "\n",
      "Copying Other Files (6 of 7): Y1_Ctrl\n",
      "\n",
      "Copying Other Files (7 of 7): Y1_PIMQ\n",
      "\n",
      "Saving ArchRProject...\n",
      "\n",
      "Loading ArchRProject...\n",
      "\n",
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Persist project with peak annotations\n",
    "proj <- saveArchRProject(ArchRProj = proj, outputDirectory = '/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_scpeakset')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (archr_env)",
   "language": "R",
   "name": "archr_env"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
