{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "673f27b4",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: rhdf5\n",
      "\n",
      "Loading required package: magrittr\n",
      "\n",
      "\n",
      "Attaching package: ‘magrittr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    subtract\n",
      "\n",
      "\n",
      "Loading required package: BSgenome\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: BiocIO\n",
      "\n",
      "Loading required package: rtracklayer\n",
      "\n",
      "\n",
      "Attaching package: ‘rtracklayer’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocIO’:\n",
      "\n",
      "    FileForFormat\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biostrings’:\n",
      "\n",
      "    collapse, intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:XVector’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "Setting default genome to Mm10.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ArchR)\n",
    "library(parallel)\n",
    "library(BSgenome.Mmusculus.UCSC.mm10)\n",
    "library(ggpubr)\n",
    "library(dplyr)\n",
    "\n",
    "addArchRGenome(\"mm10\")\n",
    "\n",
    "set.seed(147)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "38d084db",
   "metadata": {},
   "source": [
    "# D30 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8ad94cd3",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj <- loadArchRProject(\"../intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea7f3722",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Sample</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>D30_Ctrl</td><td>5403</td></tr>\n",
       "\t<tr><td>D30_PIMQ</td><td>5325</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Sample & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t D30\\_Ctrl & 5403\\\\\n",
       "\t D30\\_PIMQ & 5325\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 2\n",
       "\n",
       "| Sample &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| D30_Ctrl | 5403 |\n",
       "| D30_PIMQ | 5325 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample   n   \n",
       "1 D30_Ctrl 5403\n",
       "2 D30_PIMQ 5325"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Cell counts per sample\n",
    "data.frame(proj@cellColData) %>% group_by(Sample) %>% summarize(n = n())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "80b3d736-5440-4dbc-b0f4-fcb7ff3a03b3",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Export UMAP coordinates for SEACells initialization\n",
    "umap_coords <- proj@embeddings$UMAP$df\n",
    "dir.create(\"../intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset/seacells/\", showWarnings = FALSE, recursive = TRUE)\n",
    "write.table(umap_coords, '../intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset/seacells/D30_umap_df.txt')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "47112905",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addMotifAnnotations-dfa36f77665b-Date-2025-01-05_Time-11-43-44.681972.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 11:43:47.894986 : Gettting Motif Set, Species : Mus musculus, 0.001 mins elapsed.\n",
      "\n",
      "Using version 2 motifs!\n",
      "\n",
      "2025-01-05 11:43:49.370205 : Finding Motif Positions with motifmatchr!, 0.026 mins elapsed.\n",
      "\n",
      "2025-01-05 11:45:37.723377 : Creating Motif Overlap Matrix, 1.832 mins elapsed.\n",
      "\n",
      "2025-01-05 11:45:39.879883 : Finished Getting Motif Info!, 1.868 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addMotifAnnotations-dfa36f77665b-Date-2025-01-05_Time-11-43-44.681972.log\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving ArchRProject...\n",
      "\n",
      "Loading ArchRProject...\n",
      "\n",
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Add motif annotations using CIS-BP database and export motif-peak matching matrix\n",
    "proj <- addMotifAnnotations(ArchRProj = proj, motifSet = \"cisbp\", name = \"Motif\")\n",
    "motif_matches <- readRDS(proj@peakAnnotation$Motif$Matches)\n",
    "dir.create(\"../intermediate_outputs3/matrices/\", showWarnings = FALSE, recursive = TRUE)\n",
    "writeMM(assay(motif_matches), \"../intermediate_outputs3/matrices/D30_scpeakSet_MotifMatrix.mtx\")\n",
    "proj <- saveArchRProject(ArchRProj = proj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d587bbb1",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(\"seacells\"):\n",
      "“'seacells' already exists”\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-dfa32a374f57-Date-2025-01-05_Time-11-57-01.067424.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 11:57:12.268919 : Organizing colData, 0.187 mins elapsed.\n",
      "\n",
      "2025-01-05 11:57:12.297902 : Organizing rowData, 0.187 mins elapsed.\n",
      "\n",
      "2025-01-05 11:57:12.300601 : Organizing rowRanges, 0.187 mins elapsed.\n",
      "\n",
      "2025-01-05 11:57:12.304919 : Organizing Assays (1 of 1), 0.187 mins elapsed.\n",
      "\n",
      "2025-01-05 11:57:12.426687 : Constructing SummarizedExperiment, 0.189 mins elapsed.\n",
      "\n",
      "2025-01-05 11:57:13.08588 : Finished Matrix Creation, 0.2 mins elapsed.\n",
      "\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 1.9 GiB”\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-dfa341950a9b-Date-2025-01-05_Time-12-00-31.640148.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 12:00:47.693868 : Organizing colData, 0.268 mins elapsed.\n",
      "\n",
      "2025-01-05 12:00:47.72267 : Organizing rowData, 0.268 mins elapsed.\n",
      "\n",
      "2025-01-05 12:00:47.726878 : Organizing rowRanges, 0.268 mins elapsed.\n",
      "\n",
      "2025-01-05 12:00:47.7331 : Organizing Assays (1 of 1), 0.268 mins elapsed.\n",
      "\n",
      "2025-01-05 12:00:47.958558 : Constructing SummarizedExperiment, 0.272 mins elapsed.\n",
      "\n",
      "2025-01-05 12:00:48.650455 : Finished Matrix Creation, 0.284 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Export data for SEACells metacell construction\n",
    "\n",
    "# Set up export directories\n",
    "setwd(getOutputDirectory(proj))\n",
    "dir.create('seacells')\n",
    "dir.create(\"seacells/SEACells_export\")\n",
    "setwd(\"seacells/SEACells_export\")\n",
    "\n",
    "# Export LSI embeddings for SEACells kernel construction\n",
    "write.csv(getReducedDims(proj, reducedDims = \"IterativeLSI\"), \"svd.csv\", quote = FALSE)\n",
    "write.csv(getCellColData(proj), \"cell_metadata.csv\", quote = FALSE)\n",
    "\n",
    "# Export gene activity scores\n",
    "gene_score_mat <- getMatrixFromProject(proj, useMatrix = \"GeneScoreMatrix\")\n",
    "gene_scores <- assays(gene_score_mat)[[\"GeneScoreMatrix\"]]\n",
    "gene_scores <- as.matrix(gene_scores)\n",
    "rownames(gene_scores) <- rowData(gene_score_mat)$name\n",
    "write.csv(gene_scores, \"gene_scores.csv\", quote = FALSE)\n",
    "\n",
    "# Export peak accessibility counts\n",
    "peak_set <- getPeakSet(proj)\n",
    "peak_mat <- getMatrixFromProject(proj, \"PeakMatrix\")\n",
    "\n",
    "# Reorder peaks by chromosome for consistent downstream processing\n",
    "chr_order <- sort(seqlevels(peak_set))\n",
    "peaks_by_chr <- list()\n",
    "for(chr in chr_order){ peaks_by_chr[[chr]] = peak_set[seqnames(peak_set) == chr] }\n",
    "peaks_reordered <- Reduce(\"c\", peaks_by_chr)\n",
    "\n",
    "# Export peak count matrix and feature metadata\n",
    "wd <- getwd()\n",
    "dir.create(\"peak_counts\")\n",
    "setwd(\"peak_counts\")\n",
    "\n",
    "peak_counts <- assays(peak_mat)[[\"PeakMatrix\"]]\n",
    "writeMM(peak_counts, \"counts.mtx\")\n",
    "write.csv(colnames(peak_mat), \"cells.csv\", quote = FALSE)\n",
    "names(peaks_reordered) <- sprintf(\"Peak%d\", 1:length(peaks_reordered))\n",
    "write.csv(as.data.frame(peaks_reordered), \"peaks.csv\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "188e0b40",
   "metadata": {},
   "source": [
    "# Y1 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0ba9b7b-5f0b-43f2-9c80-bc615a5b3072",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/Scripts'"
      ],
      "text/latex": [
       "'/lila/data/peer/sotougl/Fuchs/inflammatory\\_memory/scATAC/Scripts'"
      ],
      "text/markdown": [
       "'/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/Scripts'"
      ],
      "text/plain": [
       "[1] \"/lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/Scripts\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Reset working directory for Y1 dataset processing\n",
    "setwd('/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/Scripts')\n",
    "getwd()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "5be5913f",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj <- loadArchRProject(\"../intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_scpeakset\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86c3465e-300c-447c-bf89-c18ad0e0df13",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Export UMAP coordinates for SEACells initialization\n",
    "umap_coords <- proj@embeddings$UMAP$df\n",
    "dir.create(\"../intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_scpeakset/seacells/\", showWarnings = FALSE, recursive = TRUE)\n",
    "\n",
    "write.table(umap_coords, '../intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_scpeakset/seacells/Y1_umap_df.txt')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e84cb4b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Sample</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Y1_Ctrl</td><td>2182</td></tr>\n",
       "\t<tr><td>Y1_PIMQ</td><td>2294</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Sample & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Y1\\_Ctrl & 2182\\\\\n",
       "\t Y1\\_PIMQ & 2294\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 2\n",
       "\n",
       "| Sample &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Y1_Ctrl | 2182 |\n",
       "| Y1_PIMQ | 2294 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample  n   \n",
       "1 Y1_Ctrl 2182\n",
       "2 Y1_PIMQ 2294"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Cell counts per sample\n",
    "data.frame(proj@cellColData) %>% group_by(Sample) %>% summarize(n = n())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "818d9b22",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addMotifAnnotations-dfa3407b2d9-Date-2025-01-05_Time-12-03-22.331064.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 12:03:22.417368 : Gettting Motif Set, Species : Mus musculus, 0.001 mins elapsed.\n",
      "\n",
      "Using version 2 motifs!\n",
      "\n",
      "2025-01-05 12:03:24.516201 : Finding Motif Positions with motifmatchr!, 0.036 mins elapsed.\n",
      "\n",
      "2025-01-05 12:05:48.418801 : Creating Motif Overlap Matrix, 2.435 mins elapsed.\n",
      "\n",
      "2025-01-05 12:05:50.558787 : Finished Getting Motif Info!, 2.47 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addMotifAnnotations-dfa3407b2d9-Date-2025-01-05_Time-12-03-22.331064.log\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Saving ArchRProject...\n",
      "\n",
      "Loading ArchRProject...\n",
      "\n",
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Add motif annotations using CIS-BP database and export motif-peak matching matrix\n",
    "proj <- addMotifAnnotations(ArchRProj = proj, motifSet = \"cisbp\", name = \"Motif\")\n",
    "motif_matches <- readRDS(proj@peakAnnotation$Motif$Matches)\n",
    "writeMM(assay(motif_matches), \"../intermediate_outputs3/matrices/Y1_scpeakSet_MotifMatrix.mtx\")\n",
    "proj <- saveArchRProject(ArchRProj = proj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3e89333",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(\"seacells\"):\n",
      "“'seacells' already exists”\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-dfa32063d9d7-Date-2025-01-05_Time-12-05-57.601032.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 12:06:16.630113 : Organizing colData, 0.317 mins elapsed.\n",
      "\n",
      "2025-01-05 12:06:16.658039 : Organizing rowData, 0.318 mins elapsed.\n",
      "\n",
      "2025-01-05 12:06:16.660766 : Organizing rowRanges, 0.318 mins elapsed.\n",
      "\n",
      "2025-01-05 12:06:16.665023 : Organizing Assays (1 of 1), 0.318 mins elapsed.\n",
      "\n",
      "2025-01-05 12:06:16.78604 : Constructing SummarizedExperiment, 0.32 mins elapsed.\n",
      "\n",
      "2025-01-05 12:06:17.497175 : Finished Matrix Creation, 0.332 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-dfa3519afa54-Date-2025-01-05_Time-12-07-46.503544.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 12:08:11.146295 : Organizing colData, 0.411 mins elapsed.\n",
      "\n",
      "2025-01-05 12:08:11.174596 : Organizing rowData, 0.411 mins elapsed.\n",
      "\n",
      "2025-01-05 12:08:11.178828 : Organizing rowRanges, 0.411 mins elapsed.\n",
      "\n",
      "2025-01-05 12:08:11.185143 : Organizing Assays (1 of 1), 0.411 mins elapsed.\n",
      "\n",
      "2025-01-05 12:08:11.494396 : Constructing SummarizedExperiment, 0.417 mins elapsed.\n",
      "\n",
      "2025-01-05 12:08:12.173369 : Finished Matrix Creation, 0.428 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Export data for SEACells metacell construction\n",
    "\n",
    "# Set up export directories\n",
    "setwd(getOutputDirectory(proj))\n",
    "dir.create(\"seacells\")\n",
    "dir.create(\"seacells/SEACells_export\")\n",
    "setwd(\"seacells/SEACells_export\")\n",
    "\n",
    "# Export LSI embeddings for SEACells kernel construction\n",
    "write.csv(getReducedDims(proj, reducedDims = \"IterativeLSI\"), \"svd.csv\", quote = FALSE)\n",
    "write.csv(getCellColData(proj), \"cell_metadata.csv\", quote = FALSE)\n",
    "\n",
    "# Export gene activity scores\n",
    "gene_score_mat <- getMatrixFromProject(proj, useMatrix = \"GeneScoreMatrix\")\n",
    "gene_scores <- assays(gene_score_mat)[[\"GeneScoreMatrix\"]]\n",
    "gene_scores <- as.matrix(gene_scores)\n",
    "rownames(gene_scores) <- rowData(gene_score_mat)$name\n",
    "write.csv(gene_scores, \"gene_scores.csv\", quote = FALSE)\n",
    "\n",
    "# Export peak accessibility counts\n",
    "peak_set <- getPeakSet(proj)\n",
    "peak_mat <- getMatrixFromProject(proj, \"PeakMatrix\")\n",
    "\n",
    "# Reorder peaks by chromosome for consistent downstream processing\n",
    "chr_order <- sort(seqlevels(peak_set))\n",
    "peaks_by_chr <- list()\n",
    "for(chr in chr_order){ peaks_by_chr[[chr]] = peak_set[seqnames(peak_set) == chr] }\n",
    "peaks_reordered <- Reduce(\"c\", peaks_by_chr)\n",
    "\n",
    "# Export peak count matrix and feature metadata\n",
    "wd <- getwd()\n",
    "dir.create(\"peak_counts\")\n",
    "setwd(\"peak_counts\")\n",
    "\n",
    "peak_counts <- assays(peak_mat)[[\"PeakMatrix\"]]\n",
    "writeMM(peak_counts, \"counts.mtx\")\n",
    "write.csv(colnames(peak_mat), \"cells.csv\", quote = FALSE)\n",
    "names(peaks_reordered) <- sprintf(\"Peak%d\", 1:length(peaks_reordered))\n",
    "write.csv(as.data.frame(peaks_reordered), \"peaks.csv\", quote = FALSE)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (archr_env)",
   "language": "R",
   "name": "archr_env"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
