{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69957dc7-8c40-40a4-97d3-d22ea7c62764",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,\n",
      "    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,\n",
      "    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,\n",
      "    table, tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: rhdf5\n",
      "\n",
      "Loading required package: magrittr\n",
      "\n",
      "\n",
      "Attaching package: ‘magrittr’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    subtract\n",
      "\n",
      "\n",
      "Loading required package: BSgenome\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: BiocIO\n",
      "\n",
      "Loading required package: rtracklayer\n",
      "\n",
      "\n",
      "Attaching package: ‘rtracklayer’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:BiocIO’:\n",
      "\n",
      "    FileForFormat\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biostrings’:\n",
      "\n",
      "    collapse, intersect, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:XVector’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomeInfoDb’:\n",
      "\n",
      "    intersect\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, union\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:matrixStats’:\n",
      "\n",
      "    count\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in library(writexl): there is no package called ‘writexl’\n",
     "output_type": "error",
     "traceback": [
      "Error in library(writexl): there is no package called ‘writexl’\nTraceback:\n",
      "1. library(writexl)"
     ]
    }
   ],
   "source": [
    "# Load required libraries for ArchR scATAC-seq analysis\n",
    "library(ArchR)\n",
    "library(parallel)\n",
    "library(BSgenome.Mmusculus.UCSC.mm10)\n",
    "library(ggpubr)\n",
    "library(dplyr)\n",
    "library(writexl)\n",
    "\n",
    "# Set mouse mm10 genome for ArchR\n",
    "addArchRGenome(\"mm10\")\n",
    "\n",
    "# Set random seed for reproducibility\n",
    "set.seed(147)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "013a41d3-c4fe-4357-b914-b08bc206638f",
   "metadata": {},
   "source": [
    "# D30 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a98a0aa-22a2-446b-ae8d-0edfe41aa347",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Load D30 ArchR project with single-cell peak set\n",
    "proj <- loadArchRProject(\"/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_scpeakset\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "574b2ac3-e562-4070-a163-1a11a7845e19",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in min(abs(c(input%%1, input%%1 - 1)), na.rm = TRUE):\n",
      "“no non-missing arguments to min; returning Inf”\n",
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-16242cda391b-Date-2025-01-05_Time-13-02-00.001141.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:02:00.456233 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      ".createArrowGroup : Arrow Group already exists! Dropping Group from ArrowFile! This will take ~10-30 seconds!\n",
      "\n",
      ".dropGroupsFromArrow : Initializing Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding Metadata to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding SubGroups to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Move Temp ArrowFile to ArrowFile\n",
      "\n",
      "2025-01-05 13:02:21.78533 : Adding D30_PIMQ to PeakMatrix for Chr (1 of 20)!, 0.014 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:28.126121 : Adding D30_PIMQ to PeakMatrix for Chr (2 of 20)!, 0.119 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:34.814415 : Adding D30_PIMQ to PeakMatrix for Chr (3 of 20)!, 0.231 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:38.697737 : Adding D30_PIMQ to PeakMatrix for Chr (4 of 20)!, 0.296 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:43.227921 : Adding D30_PIMQ to PeakMatrix for Chr (5 of 20)!, 0.371 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:47.457886 : Adding D30_PIMQ to PeakMatrix for Chr (6 of 20)!, 0.442 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:51.739634 : Adding D30_PIMQ to PeakMatrix for Chr (7 of 20)!, 0.513 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:56.169861 : Adding D30_PIMQ to PeakMatrix for Chr (8 of 20)!, 0.587 mins elapsed.\n",
      "\n",
      "2025-01-05 13:02:59.973855 : Adding D30_PIMQ to PeakMatrix for Chr (9 of 20)!, 0.65 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:03.996098 : Adding D30_PIMQ to PeakMatrix for Chr (10 of 20)!, 0.717 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:07.818157 : Adding D30_PIMQ to PeakMatrix for Chr (11 of 20)!, 0.781 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:13.26617 : Adding D30_PIMQ to PeakMatrix for Chr (12 of 20)!, 0.872 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:16.543972 : Adding D30_PIMQ to PeakMatrix for Chr (13 of 20)!, 0.926 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:20.029152 : Adding D30_PIMQ to PeakMatrix for Chr (14 of 20)!, 0.984 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:23.064353 : Adding D30_PIMQ to PeakMatrix for Chr (15 of 20)!, 1.035 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:26.502515 : Adding D30_PIMQ to PeakMatrix for Chr (16 of 20)!, 1.092 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:29.554691 : Adding D30_PIMQ to PeakMatrix for Chr (17 of 20)!, 1.143 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:32.969419 : Adding D30_PIMQ to PeakMatrix for Chr (18 of 20)!, 1.2 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:35.750529 : Adding D30_PIMQ to PeakMatrix for Chr (19 of 20)!, 1.247 mins elapsed.\n",
      "\n",
      "2025-01-05 13:03:38.544856 : Adding D30_PIMQ to PeakMatrix for Chr (20 of 20)!, 1.293 mins elapsed.\n",
      "\n",
      ".createArrowGroup : Arrow Group already exists! Dropping Group from ArrowFile! This will take ~10-30 seconds!\n",
      "\n",
      ".dropGroupsFromArrow : Initializing Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding Metadata to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding SubGroups to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Move Temp ArrowFile to ArrowFile\n",
      "\n",
      "2025-01-05 13:04:00.234636 : Adding D30_Ctrl to PeakMatrix for Chr (1 of 20)!, 0.014 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:04.203437 : Adding D30_Ctrl to PeakMatrix for Chr (2 of 20)!, 0.08 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:08.520235 : Adding D30_Ctrl to PeakMatrix for Chr (3 of 20)!, 0.152 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:11.851201 : Adding D30_Ctrl to PeakMatrix for Chr (4 of 20)!, 0.207 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:16.268796 : Adding D30_Ctrl to PeakMatrix for Chr (5 of 20)!, 0.281 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:19.853076 : Adding D30_Ctrl to PeakMatrix for Chr (6 of 20)!, 0.341 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:23.20744 : Adding D30_Ctrl to PeakMatrix for Chr (7 of 20)!, 0.397 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:26.97885 : Adding D30_Ctrl to PeakMatrix for Chr (8 of 20)!, 0.459 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:30.290759 : Adding D30_Ctrl to PeakMatrix for Chr (9 of 20)!, 0.515 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:33.689808 : Adding D30_Ctrl to PeakMatrix for Chr (10 of 20)!, 0.571 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:36.965414 : Adding D30_Ctrl to PeakMatrix for Chr (11 of 20)!, 0.626 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:40.956105 : Adding D30_Ctrl to PeakMatrix for Chr (12 of 20)!, 0.692 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:43.805551 : Adding D30_Ctrl to PeakMatrix for Chr (13 of 20)!, 0.74 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:46.782205 : Adding D30_Ctrl to PeakMatrix for Chr (14 of 20)!, 0.79 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:49.487897 : Adding D30_Ctrl to PeakMatrix for Chr (15 of 20)!, 0.835 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:52.440058 : Adding D30_Ctrl to PeakMatrix for Chr (16 of 20)!, 0.884 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:55.130908 : Adding D30_Ctrl to PeakMatrix for Chr (17 of 20)!, 0.929 mins elapsed.\n",
      "\n",
      "2025-01-05 13:04:58.100389 : Adding D30_Ctrl to PeakMatrix for Chr (18 of 20)!, 0.978 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:00.625513 : Adding D30_Ctrl to PeakMatrix for Chr (19 of 20)!, 1.02 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:03.004693 : Adding D30_Ctrl to PeakMatrix for Chr (20 of 20)!, 1.06 mins elapsed.\n",
      "\n",
      "Overriding previous entry for ReadsInPeaks\n",
      "\n",
      "Overriding previous entry for FRIP\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-16242cda391b-Date-2025-01-05_Time-13-02-00.001141.log\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-1624270f019af-Date-2025-01-05_Time-13-05-05.269719.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:05:30.207994 : Organizing colData, 0.416 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:30.239503 : Organizing rowData, 0.416 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:30.245262 : Organizing rowRanges, 0.416 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:30.253722 : Organizing Assays (1 of 1), 0.416 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:30.540607 : Constructing SummarizedExperiment, 0.421 mins elapsed.\n",
      "\n",
      "2025-01-05 13:05:31.124734 : Finished Matrix Creation, 0.431 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(rtracklayer)\n",
    "library(GenomicRanges)\n",
    "\n",
    "# Import non-redundant peak set from bulk ATAC-seq analysis\n",
    "bedData1 <- import.bed(\"/data/peer/sotougl/final_outs/peak_sets/all_nR_BWA_MACS3_replicable_D6D30M12_peaks.bed\")\n",
    "\n",
    "grangesData <- bedData1\n",
    "\n",
    "# Filter for standard chromosomes only\n",
    "grangesData <- grangesData[grep(\"^chr\", seqnames(grangesData))]\n",
    "\n",
    "# Exclude mitochondrial and Y chromosome peaks\n",
    "grangesData <- grangesData[!(seqnames(grangesData) %in% c(\"chrM\", \"chrY\"))]\n",
    "\n",
    "# Add custom non-redundant peak set to ArchR project\n",
    "proj = addPeakSet(ArchRProj = proj, peakSet = grangesData, force = TRUE)\n",
    "\n",
    "# Disable HDF5 file locking to prevent multi-process conflicts\n",
    "Sys.setenv(RHDF5_USE_FILE_LOCKING=\"FALSE\")\n",
    "Sys.setenv(HDF5_USE_FILE_LOCKING=\"FALSE\")\n",
    "\n",
    "# Generate peak matrix by counting fragments in each peak region\n",
    "proj2 <- addPeakMatrix(proj)\n",
    "peakm2 = getMatrixFromProject(proj2, useMatrix = 'PeakMatrix')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "def1563a-19cb-465f-a9fa-92c3c0da8462",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Copying ArchRProject to new outputDirectory : /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_nonredundantpeakset\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 2)\n",
      "\n",
      "Copying Arrow Files (2 of 2)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n",
      "Copying Other Files (1 of 9): Annotations\n",
      "\n",
      "Copying Other Files (2 of 9): D30_Ctrl\n",
      "\n",
      "Copying Other Files (3 of 9): D30_PIMQ\n",
      "\n",
      "Copying Other Files (4 of 9): Embeddings\n",
      "\n",
      "Copying Other Files (5 of 9): GroupCoverages\n",
      "\n",
      "Copying Other Files (6 of 9): IterativeLSI\n",
      "\n",
      "Copying Other Files (7 of 9): IterativeLSI2\n",
      "\n",
      "Copying Other Files (8 of 9): PeakCalls\n",
      "\n",
      "Copying Other Files (9 of 9): seacells\n",
      "\n",
      "Saving ArchRProject...\n",
      "\n",
      "Loading ArchRProject...\n",
      "\n",
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Save updated D30 project with non-redundant peak set\n",
    "proj2 <- saveArchRProject(ArchRProj = proj2, outputDirectory = '/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_nonredundantpeakset')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "072c5ce3-e7b6-4484-85eb-473c6640e6c2",
   "metadata": {},
   "source": [
    "# SEACell information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9c3c403e-b661-4c41-be30-489d7caaf3a6",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Set working directory to scripts folder\n",
    "setwd('/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/Scripts')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "549ca7cf-6803-44b6-95e1-559f6179e2e6",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Load D30 project with non-redundant peak set\n",
    "proj <- loadArchRProject(\"/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_D30_min_NA_max_600_nonredundantpeakset\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b93048c-7c2b-4648-be4c-32db4ef4d219",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Sample</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>D30_Ctrl</td><td>5403</td></tr>\n",
       "\t<tr><td>D30_PIMQ</td><td>5325</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Sample & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t D30\\_Ctrl & 5403\\\\\n",
       "\t D30\\_PIMQ & 5325\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 2\n",
       "\n",
       "| Sample &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| D30_Ctrl | 5403 |\n",
       "| D30_PIMQ | 5325 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample   n   \n",
       "1 D30_Ctrl 5403\n",
       "2 D30_PIMQ 5325"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Display cell counts per sample\n",
    "data.frame(proj@cellColData) %>% group_by(Sample) %>% summarize(n = n())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23c2c889-2474-428f-a815-c98cb7b8418a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(\"seacells\"):\n",
      "“'seacells' already exists”\n",
      "Warning message in dir.create(\"seacells/SEACells_export\"):\n",
      "“'seacells/SEACells_export' already exists”\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-162422bfd1aa1-Date-2025-01-05_Time-13-08-28.888971.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:08:39.971349 : Organizing colData, 0.185 mins elapsed.\n",
      "\n",
      "2025-01-05 13:08:40.000527 : Organizing rowData, 0.185 mins elapsed.\n",
      "\n",
      "2025-01-05 13:08:40.003897 : Organizing rowRanges, 0.185 mins elapsed.\n",
      "\n",
      "2025-01-05 13:08:40.008857 : Organizing Assays (1 of 1), 0.185 mins elapsed.\n",
      "\n",
      "2025-01-05 13:08:40.113409 : Constructing SummarizedExperiment, 0.187 mins elapsed.\n",
      "\n",
      "2025-01-05 13:08:40.795886 : Finished Matrix Creation, 0.198 mins elapsed.\n",
      "\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 1.9 GiB”\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-162421ed89ccb-Date-2025-01-05_Time-13-11-59.486985.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:12:18.16528 : Organizing colData, 0.311 mins elapsed.\n",
      "\n",
      "2025-01-05 13:12:18.194405 : Organizing rowData, 0.312 mins elapsed.\n",
      "\n",
      "2025-01-05 13:12:18.199595 : Organizing rowRanges, 0.312 mins elapsed.\n",
      "\n",
      "2025-01-05 13:12:18.206882 : Organizing Assays (1 of 1), 0.312 mins elapsed.\n",
      "\n",
      "2025-01-05 13:12:18.419231 : Constructing SummarizedExperiment, 0.316 mins elapsed.\n",
      "\n",
      "2025-01-05 13:12:19.014918 : Finished Matrix Creation, 0.325 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Export data for SEACells metacell construction in Python\n",
    "\n",
    "# Create export directories\n",
    "setwd(getOutputDirectory(proj))\n",
    "dir.create('seacells')\n",
    "dir.create(\"seacells/SEACells_export\")\n",
    "setwd(\"seacells/SEACells_export\")\n",
    "\n",
    "# Export LSI dimensionality reduction coordinates\n",
    "write.csv(getReducedDims(proj, reducedDims = \"IterativeLSI\"), \"svd.csv\", quote = FALSE)\n",
    "\n",
    "# Export cell metadata\n",
    "write.csv(getCellColData(proj), \"cell_metadata.csv\", quote = FALSE)\n",
    "\n",
    "# Export gene activity scores\n",
    "gene.scores <- getMatrixFromProject(proj, useMatrix = \"GeneScoreMatrix\")\n",
    "scores <- assays(gene.scores)[[\"GeneScoreMatrix\"]]\n",
    "scores <- as.matrix(scores)\n",
    "rownames(scores) <- rowData(gene.scores)$name\n",
    "write.csv(scores, \"gene_scores.csv\", quote = FALSE)\n",
    "\n",
    "# Extract peak count matrix\n",
    "peaks <- getPeakSet(proj)\n",
    "peak.counts <- getMatrixFromProject(proj, \"PeakMatrix\")\n",
    "\n",
    "# Reorder peaks by chromosome for consistent ordering\n",
    "chr_order <- sort(seqlevels(peaks))\n",
    "reordered_features <- list()\n",
    "for(chr in chr_order){ reordered_features[[chr]] = peaks[seqnames(peaks) == chr] }\n",
    "reordered_features <- Reduce(\"c\", reordered_features)\n",
    "\n",
    "# Export peak counts in sparse matrix format\n",
    "wd <- getwd()\n",
    "dir.create(\"peak_counts\")\n",
    "setwd(\"peak_counts\")\n",
    "\n",
    "counts <- assays(peak.counts)[[\"PeakMatrix\"]]\n",
    "writeMM(counts, \"counts.mtx\")\n",
    "write.csv(colnames(peak.counts), \"cells.csv\", quote = FALSE)\n",
    "names(reordered_features) <- sprintf(\"Peak%d\", 1:length(reordered_features))\n",
    "write.csv(as.data.frame(reordered_features), \"peaks.csv\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "66fe0792-cf5f-4ea0-b36e-f2fd527a0f2d",
   "metadata": {},
   "source": [
    "# Y1 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f4b5354-487b-4ab9-bd3a-9beb9bd4a321",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Load Year 1 ArchR project with single-cell peak set\n",
    "proj <- loadArchRProject(\"/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_scpeakset\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "530522e1-6764-46b6-a7ba-dd1694697345",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in min(abs(c(input%%1, input%%1 - 1)), na.rm = TRUE):\n",
      "“no non-missing arguments to min; returning Inf”\n",
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-162426961ebfe-Date-2025-01-05_Time-13-12-49.172279.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:12:49.29475 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      ".createArrowGroup : Arrow Group already exists! Dropping Group from ArrowFile! This will take ~10-30 seconds!\n",
      "\n",
      ".dropGroupsFromArrow : Initializing Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding Metadata to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding SubGroups to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Move Temp ArrowFile to ArrowFile\n",
      "\n",
      "2025-01-05 13:13:06.334236 : Adding Y1_PIMQ to PeakMatrix for Chr (1 of 20)!, 0.013 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:11.525617 : Adding Y1_PIMQ to PeakMatrix for Chr (2 of 20)!, 0.1 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:17.284452 : Adding Y1_PIMQ to PeakMatrix for Chr (3 of 20)!, 0.196 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:21.514076 : Adding Y1_PIMQ to PeakMatrix for Chr (4 of 20)!, 0.266 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:26.763621 : Adding Y1_PIMQ to PeakMatrix for Chr (5 of 20)!, 0.354 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:31.71074 : Adding Y1_PIMQ to PeakMatrix for Chr (6 of 20)!, 0.436 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:36.001822 : Adding Y1_PIMQ to PeakMatrix for Chr (7 of 20)!, 0.508 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:41.427078 : Adding Y1_PIMQ to PeakMatrix for Chr (8 of 20)!, 0.598 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:45.811105 : Adding Y1_PIMQ to PeakMatrix for Chr (9 of 20)!, 0.671 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:50.537911 : Adding Y1_PIMQ to PeakMatrix for Chr (10 of 20)!, 0.75 mins elapsed.\n",
      "\n",
      "2025-01-05 13:13:54.820264 : Adding Y1_PIMQ to PeakMatrix for Chr (11 of 20)!, 0.822 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:00.789763 : Adding Y1_PIMQ to PeakMatrix for Chr (12 of 20)!, 0.921 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:04.512848 : Adding Y1_PIMQ to PeakMatrix for Chr (13 of 20)!, 0.983 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:08.381055 : Adding Y1_PIMQ to PeakMatrix for Chr (14 of 20)!, 1.048 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:11.829859 : Adding Y1_PIMQ to PeakMatrix for Chr (15 of 20)!, 1.105 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:15.741094 : Adding Y1_PIMQ to PeakMatrix for Chr (16 of 20)!, 1.17 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:19.17442 : Adding Y1_PIMQ to PeakMatrix for Chr (17 of 20)!, 1.227 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:23.260053 : Adding Y1_PIMQ to PeakMatrix for Chr (18 of 20)!, 1.296 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:26.351887 : Adding Y1_PIMQ to PeakMatrix for Chr (19 of 20)!, 1.347 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:29.559373 : Adding Y1_PIMQ to PeakMatrix for Chr (20 of 20)!, 1.401 mins elapsed.\n",
      "\n",
      ".createArrowGroup : Arrow Group already exists! Dropping Group from ArrowFile! This will take ~10-30 seconds!\n",
      "\n",
      ".dropGroupsFromArrow : Initializing Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding Metadata to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Adding SubGroups to Temp ArrowFile\n",
      "\n",
      ".dropGroupsFromArrow : Move Temp ArrowFile to ArrowFile\n",
      "\n",
      "2025-01-05 13:14:48.958063 : Adding Y1_Ctrl to PeakMatrix for Chr (1 of 20)!, 0.013 mins elapsed.\n",
      "\n",
      "2025-01-05 13:14:54.197162 : Adding Y1_Ctrl to PeakMatrix for Chr (2 of 20)!, 0.101 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:00.042478 : Adding Y1_Ctrl to PeakMatrix for Chr (3 of 20)!, 0.198 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:04.289111 : Adding Y1_Ctrl to PeakMatrix for Chr (4 of 20)!, 0.269 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:09.595781 : Adding Y1_Ctrl to PeakMatrix for Chr (5 of 20)!, 0.357 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:14.533599 : Adding Y1_Ctrl to PeakMatrix for Chr (6 of 20)!, 0.44 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:18.763618 : Adding Y1_Ctrl to PeakMatrix for Chr (7 of 20)!, 0.51 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:24.21083 : Adding Y1_Ctrl to PeakMatrix for Chr (8 of 20)!, 0.601 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:28.589118 : Adding Y1_Ctrl to PeakMatrix for Chr (9 of 20)!, 0.674 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:33.354884 : Adding Y1_Ctrl to PeakMatrix for Chr (10 of 20)!, 0.753 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:37.635232 : Adding Y1_Ctrl to PeakMatrix for Chr (11 of 20)!, 0.825 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:43.68981 : Adding Y1_Ctrl to PeakMatrix for Chr (12 of 20)!, 0.926 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:47.348743 : Adding Y1_Ctrl to PeakMatrix for Chr (13 of 20)!, 0.987 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:51.213442 : Adding Y1_Ctrl to PeakMatrix for Chr (14 of 20)!, 1.051 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:54.600252 : Adding Y1_Ctrl to PeakMatrix for Chr (15 of 20)!, 1.107 mins elapsed.\n",
      "\n",
      "2025-01-05 13:15:58.553586 : Adding Y1_Ctrl to PeakMatrix for Chr (16 of 20)!, 1.173 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:02.01607 : Adding Y1_Ctrl to PeakMatrix for Chr (17 of 20)!, 1.231 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:06.079852 : Adding Y1_Ctrl to PeakMatrix for Chr (18 of 20)!, 1.299 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:09.166132 : Adding Y1_Ctrl to PeakMatrix for Chr (19 of 20)!, 1.35 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:12.39866 : Adding Y1_Ctrl to PeakMatrix for Chr (20 of 20)!, 1.404 mins elapsed.\n",
      "\n",
      "Overriding previous entry for ReadsInPeaks\n",
      "\n",
      "Overriding previous entry for FRIP\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-162426961ebfe-Date-2025-01-05_Time-13-12-49.172279.log\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-16242d95b354-Date-2025-01-05_Time-13-16-14.821435.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:16:37.893345 : Organizing colData, 0.385 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:37.922434 : Organizing rowData, 0.385 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:37.927991 : Organizing rowRanges, 0.385 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:37.935715 : Organizing Assays (1 of 1), 0.385 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:38.291319 : Constructing SummarizedExperiment, 0.391 mins elapsed.\n",
      "\n",
      "2025-01-05 13:16:38.909494 : Finished Matrix Creation, 0.401 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(rtracklayer)\n",
    "library(GenomicRanges)\n",
    "\n",
    "# Import non-redundant peak set from bulk ATAC-seq analysis\n",
    "bedData1 <- import.bed(\"/data/peer/sotougl/final_outs/peak_sets/all_nR_BWA_MACS3_replicable_D6D30M12_peaks.bed\")\n",
    "\n",
    "grangesData <- bedData1\n",
    "\n",
    "# Filter for standard chromosomes only\n",
    "grangesData <- grangesData[grep(\"^chr\", seqnames(grangesData))]\n",
    "\n",
    "# Exclude mitochondrial and Y chromosome peaks\n",
    "grangesData <- grangesData[!(seqnames(grangesData) %in% c(\"chrM\", \"chrY\"))]\n",
    "\n",
    "# Add custom non-redundant peak set to ArchR project\n",
    "proj = addPeakSet(ArchRProj = proj, peakSet = grangesData, force = TRUE)\n",
    "\n",
    "# Disable HDF5 file locking to prevent multi-process conflicts\n",
    "Sys.setenv(RHDF5_USE_FILE_LOCKING=\"FALSE\")\n",
    "Sys.setenv(HDF5_USE_FILE_LOCKING=\"FALSE\")\n",
    "\n",
    "# Generate peak matrix by counting fragments in each peak region\n",
    "proj2 <- addPeakMatrix(proj)\n",
    "peakm2 = getMatrixFromProject(proj2, useMatrix = 'PeakMatrix')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5a3573a-ceaf-4a7d-b69f-9b49cf5e7363",
   "metadata": {
    "scrolled": true,
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Copying ArchRProject to new outputDirectory : /lila/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_nonredundantpeakset\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 2)\n",
      "\n",
      "Copying Arrow Files (2 of 2)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n",
      "Copying Other Files (1 of 9): Annotations\n",
      "\n",
      "Copying Other Files (2 of 9): Embeddings\n",
      "\n",
      "Copying Other Files (3 of 9): GroupCoverages\n",
      "\n",
      "Copying Other Files (4 of 9): IterativeLSI\n",
      "\n",
      "Copying Other Files (5 of 9): IterativeLSI2\n",
      "\n",
      "Copying Other Files (6 of 9): PeakCalls\n",
      "\n",
      "Copying Other Files (7 of 9): seacells\n",
      "\n",
      "Copying Other Files (8 of 9): Y1_Ctrl\n",
      "\n",
      "Copying Other Files (9 of 9): Y1_PIMQ\n",
      "\n",
      "Saving ArchRProject...\n",
      "\n",
      "Loading ArchRProject...\n",
      "\n",
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Save updated Year 1 project with non-redundant peak set\n",
    "proj2 <- saveArchRProject(ArchRProj = proj2, outputDirectory = '/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_nonredundantpeakset')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fd1296f-dbd1-4b97-9e79-4b9beafe42ba",
   "metadata": {},
   "source": [
    "# SEACell Information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9ef83da-90a2-4550-8b34-77bbf2113f2b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# Set working directory to scripts folder\n",
    "setwd('/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/Scripts')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8bc97c99-cf2e-4594-b0e0-42ada4301e9b",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Successfully loaded ArchRProject!\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Load Year 1 project with non-redundant peak set\n",
    "proj <- loadArchRProject(\"/data/peer/sotougl/Fuchs/inflammatory_memory/scATAC/intermediate_outputs3/projects/ArchR_Y1_min_NA_max_600_nonredundantpeakset\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "64607b6b-3378-408d-b73b-50e8786fd474",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in dir.create(\"seacells\"):\n",
      "“'seacells' already exists”\n",
      "Warning message in dir.create(\"seacells/SEACells_export\"):\n",
      "“'seacells/SEACells_export' already exists”\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-162422b51e943-Date-2025-01-05_Time-13-17-57.959237.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:18:08.963493 : Organizing colData, 0.183 mins elapsed.\n",
      "\n",
      "2025-01-05 13:18:08.992688 : Organizing rowData, 0.184 mins elapsed.\n",
      "\n",
      "2025-01-05 13:18:08.996125 : Organizing rowRanges, 0.184 mins elapsed.\n",
      "\n",
      "2025-01-05 13:18:09.001036 : Organizing Assays (1 of 1), 0.184 mins elapsed.\n",
      "\n",
      "2025-01-05 13:18:09.130272 : Constructing SummarizedExperiment, 0.186 mins elapsed.\n",
      "\n",
      "2025-01-05 13:18:09.873749 : Finished Matrix Creation, 0.199 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-1624265346d9c-Date-2025-01-05_Time-13-19-38.444416.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2025-01-05 13:20:01.395921 : Organizing colData, 0.383 mins elapsed.\n",
      "\n",
      "2025-01-05 13:20:01.425358 : Organizing rowData, 0.383 mins elapsed.\n",
      "\n",
      "2025-01-05 13:20:01.430864 : Organizing rowRanges, 0.383 mins elapsed.\n",
      "\n",
      "2025-01-05 13:20:01.438654 : Organizing Assays (1 of 1), 0.383 mins elapsed.\n",
      "\n",
      "2025-01-05 13:20:01.774878 : Constructing SummarizedExperiment, 0.389 mins elapsed.\n",
      "\n",
      "2025-01-05 13:20:02.413192 : Finished Matrix Creation, 0.399 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Export data for SEACells metacell construction in Python\n",
    "\n",
    "# Create export directories\n",
    "setwd(getOutputDirectory(proj))\n",
    "dir.create('seacells')\n",
    "dir.create(\"seacells/SEACells_export\")\n",
    "setwd(\"seacells/SEACells_export\")\n",
    "\n",
    "# Export LSI dimensionality reduction coordinates\n",
    "write.csv(getReducedDims(proj, reducedDims = \"IterativeLSI\"), \"svd.csv\", quote = FALSE)\n",
    "\n",
    "# Export cell metadata\n",
    "write.csv(getCellColData(proj), \"cell_metadata.csv\", quote = FALSE)\n",
    "\n",
    "# Export gene activity scores\n",
    "gene.scores <- getMatrixFromProject(proj, useMatrix = \"GeneScoreMatrix\")\n",
    "scores <- assays(gene.scores)[[\"GeneScoreMatrix\"]]\n",
    "scores <- as.matrix(scores)\n",
    "rownames(scores) <- rowData(gene.scores)$name\n",
    "write.csv(scores, \"gene_scores.csv\", quote = FALSE)\n",
    "\n",
    "# Extract peak count matrix\n",
    "peaks <- getPeakSet(proj)\n",
    "peak.counts <- getMatrixFromProject(proj, \"PeakMatrix\")\n",
    "\n",
    "# Reorder peaks by chromosome for consistent ordering\n",
    "chr_order <- sort(seqlevels(peaks))\n",
    "reordered_features <- list()\n",
    "for(chr in chr_order){ reordered_features[[chr]] = peaks[seqnames(peaks) == chr] }\n",
    "reordered_features <- Reduce(\"c\", reordered_features)\n",
    "\n",
    "# Export peak counts in sparse matrix format\n",
    "wd <- getwd()\n",
    "dir.create(\"peak_counts\")\n",
    "setwd(\"peak_counts\")\n",
    "\n",
    "counts <- assays(peak.counts)[[\"PeakMatrix\"]]\n",
    "writeMM(counts, \"counts.mtx\")\n",
    "write.csv(colnames(peak.counts), \"cells.csv\", quote = FALSE)\n",
    "names(reordered_features) <- sprintf(\"Peak%d\", 1:length(reordered_features))\n",
    "write.csv(as.data.frame(reordered_features), \"peaks.csv\", quote = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "923208fb-23f9-4c85-9c8c-756811129c25",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 2 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Sample</th><th scope=col>n</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>Y1_Ctrl</td><td>2182</td></tr>\n",
       "\t<tr><td>Y1_PIMQ</td><td>2294</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 2 × 2\n",
       "\\begin{tabular}{ll}\n",
       " Sample & n\\\\\n",
       " <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t Y1\\_Ctrl & 2182\\\\\n",
       "\t Y1\\_PIMQ & 2294\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 2 × 2\n",
       "\n",
       "| Sample &lt;chr&gt; | n &lt;int&gt; |\n",
       "|---|---|\n",
       "| Y1_Ctrl | 2182 |\n",
       "| Y1_PIMQ | 2294 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample  n   \n",
       "1 Y1_Ctrl 2182\n",
       "2 Y1_PIMQ 2294"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Display cell counts per sample\n",
    "data.frame(proj@cellColData) %>% group_by(Sample) %>% summarize(n = n())"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R (archr_env)",
   "language": "R",
   "name": "archr_env"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
